Index: sonic-platform-modules-ingrasys/s8810-32q/modules/eeprom_mb.c
======================================================================
--- sonic-platform-modules-ingrasys.orig/s8810-32q/modules/eeprom_mb.c
+++ sonic-platform-modules-ingrasys/s8810-32q/modules/eeprom_mb.c
@@ -150,8 +150,7 @@
 	return 0;
 }
 
-static int mb_eeprom_probe(struct i2c_client *client,
-			const struct i2c_device_id *id)
+static int mb_eeprom_probe(struct i2c_client *client)
 {
 	struct eeprom_data *data;
 	int err;
@@ -179,12 +178,11 @@
 	return err;
 }
 
-static int mb_eeprom_remove(struct i2c_client *client)
+static void mb_eeprom_remove(struct i2c_client *client)
 {
 	sysfs_remove_bin_file(&client->dev.kobj, &mb_eeprom_attr);
 	kfree(i2c_get_clientdata(client));
 
-	return 0;
 }
 
 static const struct i2c_device_id mb_eeprom_id[] = {
Index: sonic-platform-modules-ingrasys/s8900-54xc/modules/eeprom_mb.c
======================================================================
--- sonic-platform-modules-ingrasys.orig/s8900-54xc/modules/eeprom_mb.c
+++ sonic-platform-modules-ingrasys/s8900-54xc/modules/eeprom_mb.c
@@ -150,8 +150,7 @@
 	return 0;
 }
 
-static int mb_eeprom_probe(struct i2c_client *client,
-			const struct i2c_device_id *id)
+static int mb_eeprom_probe(struct i2c_client *client)
 {
 	struct eeprom_data *data;
 	int err;
@@ -179,12 +178,11 @@
 	return err;
 }
 
-static int mb_eeprom_remove(struct i2c_client *client)
+static void mb_eeprom_remove(struct i2c_client *client)
 {
 	sysfs_remove_bin_file(&client->dev.kobj, &mb_eeprom_attr);
 	kfree(i2c_get_clientdata(client));
 
-	return 0;
 }
 
 static const struct i2c_device_id mb_eeprom_id[] = {
Index: sonic-platform-modules-ingrasys/s8900-64xc/modules/eeprom_mb.c
======================================================================
--- sonic-platform-modules-ingrasys.orig/s8900-64xc/modules/eeprom_mb.c
+++ sonic-platform-modules-ingrasys/s8900-64xc/modules/eeprom_mb.c
@@ -150,8 +150,7 @@
 	return 0;
 }
 
-static int mb_eeprom_probe(struct i2c_client *client,
-			const struct i2c_device_id *id)
+static int mb_eeprom_probe(struct i2c_client *client)
 {
 	struct eeprom_data *data;
 	int err;
@@ -179,12 +178,11 @@
 	return err;
 }
 
-static int mb_eeprom_remove(struct i2c_client *client)
+static void mb_eeprom_remove(struct i2c_client *client)
 {
 	sysfs_remove_bin_file(&client->dev.kobj, &mb_eeprom_attr);
 	kfree(i2c_get_clientdata(client));
 
-	return 0;
 }
 
 static const struct i2c_device_id mb_eeprom_id[] = {
Index: sonic-platform-modules-ingrasys/s9100/modules/eeprom_mb.c
======================================================================
--- sonic-platform-modules-ingrasys.orig/s9100/modules/eeprom_mb.c
+++ sonic-platform-modules-ingrasys/s9100/modules/eeprom_mb.c
@@ -150,8 +150,7 @@
 	return 0;
 }
 
-static int mb_eeprom_probe(struct i2c_client *client,
-			const struct i2c_device_id *id)
+static int mb_eeprom_probe(struct i2c_client *client)
 {
 	struct eeprom_data *data;
 	int err;
@@ -179,12 +178,11 @@
 	return err;
 }
 
-static int mb_eeprom_remove(struct i2c_client *client)
+static void mb_eeprom_remove(struct i2c_client *client)
 {
 	sysfs_remove_bin_file(&client->dev.kobj, &mb_eeprom_attr);
 	kfree(i2c_get_clientdata(client));
 
-	return 0;
 }
 
 static const struct i2c_device_id mb_eeprom_id[] = {
Index: sonic-platform-modules-ingrasys/s9200-64x/modules/eeprom_mb.c
======================================================================
--- sonic-platform-modules-ingrasys.orig/s9200-64x/modules/eeprom_mb.c
+++ sonic-platform-modules-ingrasys/s9200-64x/modules/eeprom_mb.c
@@ -205,8 +205,7 @@
     return 0;
 }
 
-static int mb_eeprom_probe(struct i2c_client *client,
-            const struct i2c_device_id *id)
+static int mb_eeprom_probe(struct i2c_client *client)
 {
     struct eeprom_data *data;
     int err;
@@ -234,12 +233,11 @@
     return err;
 }
 
-static int mb_eeprom_remove(struct i2c_client *client)
+static void mb_eeprom_remove(struct i2c_client *client)
 {
     sysfs_remove_bin_file(&client->dev.kobj, &mb_eeprom_attr);
     kfree(i2c_get_clientdata(client));
 
-    return 0;
 }
 
 static const struct i2c_device_id mb_eeprom_id[] = {
Index: sonic-platform-modules-ingrasys/s8900-54xc/modules/i2c_cpld.c
======================================================================
--- sonic-platform-modules-ingrasys.orig/s8900-54xc/modules/i2c_cpld.c
+++ sonic-platform-modules-ingrasys/s8900-54xc/modules/i2c_cpld.c
@@ -363,8 +363,7 @@
 }
 
 /* cpld drvier probe */
-static int i2c_cpld_probe(struct i2c_client *client,
-                    const struct i2c_device_id *dev_id)
+static int i2c_cpld_probe(struct i2c_client *client)
 {
     int status;
     struct cpld_data *data = NULL;
@@ -404,13 +403,12 @@
 }
 
 /* cpld drvier remove */
-static int i2c_cpld_remove(struct i2c_client *client)
+static void i2c_cpld_remove(struct i2c_client *client)
 {
 
     sysfs_remove_group(&client->dev.kobj, &i2c_cpld_group);
 
     i2c_cpld_remove_client(client);
-    return 0;
 }
 
 MODULE_DEVICE_TABLE(i2c, i2c_cpld_id);
Index: sonic-platform-modules-ingrasys/s8900-64xc/modules/i2c_cpld.c
======================================================================
--- sonic-platform-modules-ingrasys.orig/s8900-64xc/modules/i2c_cpld.c
+++ sonic-platform-modules-ingrasys/s8900-64xc/modules/i2c_cpld.c
@@ -363,8 +363,7 @@
 }
 
 /* cpld drvier probe */
-static int i2c_cpld_probe(struct i2c_client *client,
-                    const struct i2c_device_id *dev_id)
+static int i2c_cpld_probe(struct i2c_client *client)
 {
     int status;
     struct cpld_data *data = NULL;
@@ -404,13 +403,12 @@
 }
 
 /* cpld drvier remove */
-static int i2c_cpld_remove(struct i2c_client *client)
+static void i2c_cpld_remove(struct i2c_client *client)
 {
 
     sysfs_remove_group(&client->dev.kobj, &i2c_cpld_group);
 
     i2c_cpld_remove_client(client);
-    return 0;
 }
 
 MODULE_DEVICE_TABLE(i2c, i2c_cpld_id);
Index: sonic-platform-modules-ingrasys/s9100/modules/i2c_cpld.c
======================================================================
--- sonic-platform-modules-ingrasys.orig/s9100/modules/i2c_cpld.c
+++ sonic-platform-modules-ingrasys/s9100/modules/i2c_cpld.c
@@ -363,8 +363,7 @@
 }
 
 /* cpld drvier probe */
-static int i2c_cpld_probe(struct i2c_client *client,
-                    const struct i2c_device_id *dev_id)
+static int i2c_cpld_probe(struct i2c_client *client)
 {
     int status;
     struct cpld_data *data = NULL;
@@ -404,13 +403,12 @@
 }
 
 /* cpld drvier remove */
-static int i2c_cpld_remove(struct i2c_client *client)
+static void i2c_cpld_remove(struct i2c_client *client)
 {
 
     sysfs_remove_group(&client->dev.kobj, &i2c_cpld_group);
 
     i2c_cpld_remove_client(client);
-    return 0;
 }
 
 MODULE_DEVICE_TABLE(i2c, i2c_cpld_id);
Index: sonic-platform-modules-ingrasys/s9200-64x/modules/ingrasys_s9200_64x_psu.c
======================================================================
--- sonic-platform-modules-ingrasys.orig/s9200-64x/modules/ingrasys_s9200_64x_psu.c
+++ sonic-platform-modules-ingrasys/s9200-64x/modules/ingrasys_s9200_64x_psu.c
@@ -173,9 +173,9 @@
 }
 
 static int 
-s9200_psu_probe(struct i2c_client *client,
-                const struct i2c_device_id *dev_id)
+s9200_psu_probe(struct i2c_client *client)
 {
+    const struct i2c_device_id *dev_id = i2c_client_get_device_id(client);
     struct s9200_psu_data *data;
     int status, err;
 
@@ -222,15 +222,14 @@
     return status;
 }
 
-static int 
+static void 
 s9200_psu_remove(struct i2c_client *client)
 {
     struct s9200_psu_data *data = i2c_get_clientdata(client);
 
     sysfs_remove_group(&client->dev.kobj, &s9200_psu_group);
     kfree(data);
-    
-    return 0;
+
 }
 
 
Index: sonic-platform-modules-ingrasys/s8810-32q/modules/ingrasys_s8810_32q_psu.c
======================================================================
--- sonic-platform-modules-ingrasys.orig/s8810-32q/modules/ingrasys_s8810_32q_psu.c
+++ sonic-platform-modules-ingrasys/s8810-32q/modules/ingrasys_s8810_32q_psu.c
@@ -174,9 +174,9 @@
 }
 
 static int 
-s8810_psu_probe(struct i2c_client *client,
-                const struct i2c_device_id *dev_id)
+s8810_psu_probe(struct i2c_client *client)
 {
+    const struct i2c_device_id *dev_id = i2c_client_get_device_id(client);
     struct s8810_psu_data *data;
     int status, err;
 
@@ -223,15 +223,14 @@
     return status;
 }
 
-static int 
+static void 
 s8810_psu_remove(struct i2c_client *client)
 {
     struct s8810_psu_data *data = i2c_get_clientdata(client);
 
     sysfs_remove_group(&client->dev.kobj, &s8810_psu_group);
     kfree(data);
-    
-    return 0;
+
 }
 
 
Index: sonic-platform-modules-ingrasys/s9200-64x/modules/cpld.c
======================================================================
--- sonic-platform-modules-ingrasys.orig/s9200-64x/modules/cpld.c
+++ sonic-platform-modules-ingrasys/s9200-64x/modules/cpld.c
@@ -425,7 +425,7 @@
     return -ENODEV; 
 }
 
-static int __exit cpld_remove(struct platform_device *pdev)
+static void __exit cpld_remove(struct platform_device *pdev)
 {
     int i;
     struct i2c_adapter *parent = NULL;
@@ -445,7 +445,6 @@
         }
     }
 
-    return 0;
 }
 
 static struct platform_driver cpld_driver = {
Index: sonic-platform-modules-ingrasys/s8900-64xc/modules/qsfp_cpld.c
======================================================================
--- sonic-platform-modules-ingrasys.orig/s8900-64xc/modules/qsfp_cpld.c
+++ sonic-platform-modules-ingrasys/s8900-64xc/modules/qsfp_cpld.c
@@ -1589,7 +1589,7 @@
     return 0;
 }
 
-static int __exit
+static void __exit
 qsfp_device_remove(struct platform_device *pdev)
 {
     struct sfp_port_data *data = NULL;
@@ -1601,7 +1601,7 @@
 
     if (!pdata) {
         ERROR_MSG("Missing platform data\n");
-        return -ENOENT;
+        return;
     }
 
     for (i = 0; i < TOTAL_PORT_NUM; i+=CPLD_MUX_OFFSET) {
@@ -1620,7 +1620,6 @@
     }
 
 
-    return 0;
 }
 
 static int
@@ -1672,7 +1671,7 @@
     return -ENODEV;
 }
 
-static int __exit cpld_remove(struct platform_device *pdev)
+static void __exit cpld_remove(struct platform_device *pdev)
 {
     int i;
     struct i2c_adapter *parent = NULL;
@@ -1692,7 +1691,6 @@
         }
     }
 
-    return 0;
 }
 
 static struct platform_driver cpld_driver = {
