Index: sonic-platform-modules-dell/debian/control
===================================================================
--- sonic-platform-modules-dell.orig/debian/control
+++ sonic-platform-modules-dell/debian/control
@@ -2,96 +2,96 @@ Source: sonic-dell-platform-modules
 Section: main
 Priority: extra
 Maintainer: Dell Team <support@dell.com>
-Build-Depends: debhelper-compat (= 13), dh-python, python3, linux-headers-6.12.41+deb13-common-sonic
+Build-Depends: debhelper-compat (= 13), dh-python, python3, linux-sonic-headers-7.0.0-1002
 Rules-Requires-Root: no
 Standards-Version: 4.7.2
 
 Package: platform-modules-s6000
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
 
 Package: platform-modules-z9100
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
 
 Package: platform-modules-s6100
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
 
 Package: platform-modules-z9264f
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
 
 Package: platform-modules-s5212f
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
 
 Package: platform-modules-s5224f
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
 
 Package: platform-modules-s5232f
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
 
 Package: platform-modules-s5248f
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
 
 Package: platform-modules-s5448f
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
 
 Package: platform-modules-n3248te
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
 
 Package: platform-modules-z9332f
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
 
 Package: platform-modules-z9432f
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
 
 Package: platform-modules-n3248pxe
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
 
 Package: platform-modules-e3224f
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
 
 Package: platform-modules-s5296f
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
 
 Package: platform-modules-z9664f
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
 
 Package: platform-modules-s3248t
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
 
 Package: platform-modules-z9864f
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
Index: sonic-platform-modules-dell/z9864f/modules/fpga_gpio.c
===================================================================
--- sonic-platform-modules-dell.orig/z9864f/modules/fpga_gpio.c
+++ sonic-platform-modules-dell/z9864f/modules/fpga_gpio.c
@@ -46,7 +46,7 @@ static int fpga_gpio_get(struct gpio_chi
     return ret;
 }
 
-static void fpga_gpio_set(struct gpio_chip *gc,
+static int fpga_gpio_set(struct gpio_chip *gc,
                           unsigned gpio, int val)
 {
     struct fpga_gpio_chip *chip = gpiochip_get_data(gc);
@@ -66,6 +66,8 @@ static void fpga_gpio_set(struct gpio_ch
     iowrite32(wdata, chip->bar + reg);
 
     mutex_unlock(&chip->lock);
+
+    return 0;
 }
 
 int gpiodev_init(struct pci_dev *dev, struct fpga_dev *fpga)
@@ -98,4 +100,4 @@ void gpiodev_exit(struct pci_dev *dev)
 {
     struct fpga_dev *fpga = pci_get_drvdata(dev);
     gpiochip_remove(&fpga->gpio->gpio_chip);
-}
\ No newline at end of file
+}
