Index: sonic-platform-modules-quanta/debian/control
===================================================================
--- sonic-platform-modules-quanta.orig/debian/control
+++ sonic-platform-modules-quanta/debian/control
@@ -2,7 +2,7 @@ Source: sonic-quanta-platform-modules
 Section: main
 Priority: extra
 Maintainer: Chih-Pei Chang <Chih-Pei.Chang@qct.io>, Jonathan Tsai <jonathan.tsai@quantatw.com> 
-Build-Depends: debhelper-compat (= 13), dh-python, pybuild-plugin-pyproject, python3, linux-headers-6.12.41+deb13-common-sonic
+Build-Depends: debhelper-compat (= 13), dh-python, pybuild-plugin-pyproject, python3, linux-sonic-headers-7.0.0-1002
 Standards-Version: 3.9.3
 
 Package: sonic-platform-quanta-ix1b-32x
