Index: sonic-platform-modules-cel/debian/control
===================================================================
--- sonic-platform-modules-cel.orig/debian/control
+++ sonic-platform-modules-cel/debian/control
@@ -8,46 +8,46 @@ Standards-Version: 3.9.3
 
 Package: platform-modules-dx010
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
 
 
 Package: platform-modules-haliburton
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
 
 Package: platform-modules-seastone2
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as led, sfp
 
 Package: platform-modules-silverstone
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as led, sfp.
 
 Package: platform-modules-ds1000
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as led, sfp
 
 Package: platform-modules-questone2
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as led, sfp
 
 Package: platform-modules-ds3000
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as led, sfp
 
 Package: platform-modules-silverstone-v2
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as led, sfp
 
 Package: platform-modules-ds2000
 Architecture: amd64
-Depends: linux-image-6.12.41+deb13-sonic-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as led, sfp
