Index: sonic-platform-modules-cel/ds1000/modules/mc24lc64t.c
===================================================================
--- sonic-platform-modules-cel.orig/ds1000/modules/mc24lc64t.c
+++ sonic-platform-modules-cel/ds1000/modules/mc24lc64t.c
@@ -25,7 +25,7 @@ struct mc24lc64t_data {
 };
 
 static ssize_t mc24lc64t_read(struct file *filp, struct kobject *kobj,
-                            struct bin_attribute *bin_attr,
+                            const struct bin_attribute *bin_attr,
                             char *buf, loff_t off, size_t count)
 {
         struct i2c_client *client = kobj_to_i2c_client(kobj);
@@ -72,7 +72,7 @@ exit:
 }
 
 static ssize_t mc24lc64t_write (struct file *filp, struct kobject *kobj,
-                            struct bin_attribute *bin_attr,
+                            const struct bin_attribute *bin_attr,
                             char *buf, loff_t off, size_t count){
 
     struct i2c_client *client = kobj_to_i2c_client(kobj);
Index: sonic-platform-modules-cel/ds2000/modules/lpc_basecpld.c
===================================================================
--- sonic-platform-modules-cel.orig/ds2000/modules/lpc_basecpld.c
+++ sonic-platform-modules-cel/ds2000/modules/lpc_basecpld.c
@@ -241,7 +241,7 @@ static ssize_t setreg_store(struct devic
  * @return number of byte read.
  */
 static ssize_t dump_read(struct file *filp, struct kobject *kobj,
-				struct bin_attribute *attr, char *buf,
+				const struct bin_attribute *attr, char *buf,
 				loff_t off, size_t count)
 {
 	unsigned long i = 0;
Index: sonic-platform-modules-cel/ds2000/modules/mc24lc64t.c
===================================================================
--- sonic-platform-modules-cel.orig/ds2000/modules/mc24lc64t.c
+++ sonic-platform-modules-cel/ds2000/modules/mc24lc64t.c
@@ -25,7 +25,7 @@ struct mc24lc64t_data {
 };
 
 static ssize_t mc24lc64t_read(struct file *filp, struct kobject *kobj,
-                            struct bin_attribute *bin_attr,
+                            const struct bin_attribute *bin_attr,
                             char *buf, loff_t off, size_t count)
 {
         struct i2c_client *client = kobj_to_i2c_client(kobj);
@@ -72,7 +72,7 @@ exit:
 }
 
 static ssize_t mc24lc64t_write (struct file *filp, struct kobject *kobj,
-                            struct bin_attribute *bin_attr,
+                            const struct bin_attribute *bin_attr,
                             char *buf, loff_t off, size_t count){
 
     struct i2c_client *client = kobj_to_i2c_client(kobj);
Index: sonic-platform-modules-cel/ds2000/modules/pddf_custom_fpga_extend.c
===================================================================
--- sonic-platform-modules-cel.orig/ds2000/modules/pddf_custom_fpga_extend.c
+++ sonic-platform-modules-cel/ds2000/modules/pddf_custom_fpga_extend.c
@@ -187,7 +187,7 @@ static ssize_t set_fpga_reg_value(struct
  * @return       number of bytes read, or an error code
  */
 static ssize_t dump_read(struct file *filp, struct kobject *kobj,
-                         struct bin_attribute *attr, char *buf,
+                         const struct bin_attribute *attr, char *buf,
                          loff_t off, size_t count)
 {
     unsigned long i = 0;
Index: sonic-platform-modules-cel/ds3000/modules/mc24lc64t.c
===================================================================
--- sonic-platform-modules-cel.orig/ds3000/modules/mc24lc64t.c
+++ sonic-platform-modules-cel/ds3000/modules/mc24lc64t.c
@@ -25,7 +25,7 @@ struct mc24lc64t_data {
 };
 
 static ssize_t mc24lc64t_read(struct file *filp, struct kobject *kobj,
-                            struct bin_attribute *bin_attr,
+                            const struct bin_attribute *bin_attr,
                             char *buf, loff_t off, size_t count)
 {
         struct i2c_client *client = kobj_to_i2c_client(kobj);
@@ -73,7 +73,7 @@ exit:
 
 
 static ssize_t mc24lc64t_write (struct file *filp, struct kobject *kobj,
-                            struct bin_attribute *bin_attr,
+                            const struct bin_attribute *bin_attr,
                             char *buf, loff_t off, size_t count){
 
     struct i2c_client *client = kobj_to_i2c_client(kobj);
Index: sonic-platform-modules-cel/ds3000/modules/pddf_custom_fpga_extend.c
===================================================================
--- sonic-platform-modules-cel.orig/ds3000/modules/pddf_custom_fpga_extend.c
+++ sonic-platform-modules-cel/ds3000/modules/pddf_custom_fpga_extend.c
@@ -187,7 +187,7 @@ static ssize_t set_fpga_reg_value(struct
  * @return       number of bytes read, or an error code
  */
 static ssize_t dump_read(struct file *filp, struct kobject *kobj,
-                         struct bin_attribute *attr, char *buf,
+                         const struct bin_attribute *attr, char *buf,
                          loff_t off, size_t count)
 {
     unsigned long i = 0;
Index: sonic-platform-modules-cel/dx010/modules/mc24lc64t.c
===================================================================
--- sonic-platform-modules-cel.orig/dx010/modules/mc24lc64t.c
+++ sonic-platform-modules-cel/dx010/modules/mc24lc64t.c
@@ -24,7 +24,7 @@ struct mc24lc64t_data {
 };
 
 static ssize_t mc24lc64t_read(struct file *filp, struct kobject *kobj,
-                            struct bin_attribute *bin_attr,
+                            const struct bin_attribute *bin_attr,
                             char *buf, loff_t off, size_t count)
 {
         struct i2c_client *client = kobj_to_i2c_client(kobj);
Index: sonic-platform-modules-cel/haliburton/modules/mc24lc64t.c
===================================================================
--- sonic-platform-modules-cel.orig/haliburton/modules/mc24lc64t.c
+++ sonic-platform-modules-cel/haliburton/modules/mc24lc64t.c
@@ -24,7 +24,7 @@ struct mc24lc64t_data {
 };
 
 static ssize_t mc24lc64t_read(struct file *filp, struct kobject *kobj,
-                            struct bin_attribute *bin_attr,
+                            const struct bin_attribute *bin_attr,
                             char *buf, loff_t off, size_t count)
 {
         struct i2c_client *client = kobj_to_i2c_client(kobj);
Index: sonic-platform-modules-cel/questone2/modules/mc24lc64t.c
===================================================================
--- sonic-platform-modules-cel.orig/questone2/modules/mc24lc64t.c
+++ sonic-platform-modules-cel/questone2/modules/mc24lc64t.c
@@ -25,7 +25,7 @@ struct mc24lc64t_data {
 };
 
 static ssize_t mc24lc64t_read(struct file *filp, struct kobject *kobj,
-                            struct bin_attribute *bin_attr,
+                            const struct bin_attribute *bin_attr,
                             char *buf, loff_t off, size_t count)
 {
         struct i2c_client *client = kobj_to_i2c_client(kobj);
@@ -72,7 +72,7 @@ exit:
 }
 
 static ssize_t mc24lc64t_write (struct file *filp, struct kobject *kobj,
-                            struct bin_attribute *bin_attr,
+                            const struct bin_attribute *bin_attr,
                             char *buf, loff_t off, size_t count){
 
     struct i2c_client *client = kobj_to_i2c_client(kobj);
Index: sonic-platform-modules-cel/questone2/modules/questone2_baseboard_cpld.c
===================================================================
--- sonic-platform-modules-cel.orig/questone2/modules/questone2_baseboard_cpld.c
+++ sonic-platform-modules-cel/questone2/modules/questone2_baseboard_cpld.c
@@ -173,7 +173,7 @@ static DEVICE_ATTR_WO(setreg);
  * @return number of byte read.
  */
 static ssize_t dump_read(struct file *filp, struct kobject *kobj,
-                struct bin_attribute *attr, char *buf,
+                const struct bin_attribute *attr, char *buf,
                 loff_t off, size_t count)
 {
     unsigned long i=0;
Index: sonic-platform-modules-cel/questone2/modules/questone2_switchboard.c
===================================================================
--- sonic-platform-modules-cel.orig/questone2/modules/questone2_switchboard.c
+++ sonic-platform-modules-cel/questone2/modules/questone2_switchboard.c
@@ -587,7 +587,7 @@ static ssize_t set_fpga_reg_value(struct
  * @return       [description]
  */
 static ssize_t dump_read(struct file *filp, struct kobject *kobj,
-                struct bin_attribute *attr, char *buf,
+                const struct bin_attribute *attr, char *buf,
                 loff_t off, size_t count)
 {
     unsigned long i=0;
Index: sonic-platform-modules-cel/seastone2/modules/mc24lc64t.c
===================================================================
--- sonic-platform-modules-cel.orig/seastone2/modules/mc24lc64t.c
+++ sonic-platform-modules-cel/seastone2/modules/mc24lc64t.c
@@ -25,7 +25,7 @@ struct mc24lc64t_data {
 };
 
 static ssize_t mc24lc64t_read(struct file *filp, struct kobject *kobj,
-                            struct bin_attribute *bin_attr,
+                            const struct bin_attribute *bin_attr,
                             char *buf, loff_t off, size_t count)
 {
         struct i2c_client *client = kobj_to_i2c_client(kobj);
@@ -73,7 +73,7 @@ exit:
 
 
 static ssize_t mc24lc64t_write (struct file *filp, struct kobject *kobj,
-                            struct bin_attribute *bin_attr,
+                            const struct bin_attribute *bin_attr,
                             char *buf, loff_t off, size_t count){
 
     struct i2c_client *client = kobj_to_i2c_client(kobj);
Index: sonic-platform-modules-cel/silverstone-v2/modules/pddf_custom_fpga_extend.c
===================================================================
--- sonic-platform-modules-cel.orig/silverstone-v2/modules/pddf_custom_fpga_extend.c
+++ sonic-platform-modules-cel/silverstone-v2/modules/pddf_custom_fpga_extend.c
@@ -185,7 +185,7 @@ static ssize_t set_fpga_reg_value(struct
  * @return       number of bytes read, or an error code
  */
 static ssize_t dump_read(struct file *filp, struct kobject *kobj,
-                         struct bin_attribute *attr, char *buf,
+                         const struct bin_attribute *attr, char *buf,
                          loff_t off, size_t count)
 {
     unsigned long i = 0;
Index: sonic-platform-modules-cel/silverstone-v2/modules/pddf_custom_lpc_basecpld.c
===================================================================
--- sonic-platform-modules-cel.orig/silverstone-v2/modules/pddf_custom_lpc_basecpld.c
+++ sonic-platform-modules-cel/silverstone-v2/modules/pddf_custom_lpc_basecpld.c
@@ -193,7 +193,7 @@ static DEVICE_ATTR_WO(setreg);
  * @return number of byte read.
  */
 static ssize_t dump_read(struct file *filp, struct kobject *kobj,
-                struct bin_attribute *attr, char *buf,
+                const struct bin_attribute *attr, char *buf,
                 loff_t off, size_t count)
 {
     unsigned long i=0;
Index: sonic-platform-modules-cel/silverstone/modules/baseboard-lpc.c
===================================================================
--- sonic-platform-modules-cel.orig/silverstone/modules/baseboard-lpc.c
+++ sonic-platform-modules-cel/silverstone/modules/baseboard-lpc.c
@@ -185,7 +185,7 @@ static DEVICE_ATTR_WO(setreg);
  * Read all CPLD register in binary mode.
  */
 static ssize_t dump_read(struct file *filp, struct kobject *kobj,
-                struct bin_attribute *attr, char *buf,
+                const struct bin_attribute *attr, char *buf,
                 loff_t off, size_t count)
 {
     unsigned long i=0;
Index: sonic-platform-modules-cel/silverstone/modules/mc24lc64t.c
===================================================================
--- sonic-platform-modules-cel.orig/silverstone/modules/mc24lc64t.c
+++ sonic-platform-modules-cel/silverstone/modules/mc24lc64t.c
@@ -25,7 +25,7 @@ struct mc24lc64t_data {
 };
 
 static ssize_t mc24lc64t_read(struct file *filp, struct kobject *kobj,
-                            struct bin_attribute *bin_attr,
+                            const struct bin_attribute *bin_attr,
                             char *buf, loff_t off, size_t count)
 {
         struct i2c_client *client = kobj_to_i2c_client(kobj);
@@ -73,7 +73,7 @@ exit:
 
 
 static ssize_t mc24lc64t_write (struct file *filp, struct kobject *kobj,
-                            struct bin_attribute *bin_attr,
+                            const struct bin_attribute *bin_attr,
                             char *buf, loff_t off, size_t count){
 
     struct i2c_client *client = kobj_to_i2c_client(kobj);
