Index: sonic-platform-modules-cel/silverstone-v2/modules/tps536c7.c
===================================================================
--- sonic-platform-modules-cel.orig/silverstone-v2/modules/tps536c7.c
+++ sonic-platform-modules-cel/silverstone-v2/modules/tps536c7.c
@@ -130,4 +130,4 @@ module_i2c_driver(tps536c7_driver);
 MODULE_AUTHOR("Vadim Pasternak <vadimp@mellanox.com>");
 MODULE_DESCRIPTION("PMBus driver for Texas Instruments TPS536C7");
 MODULE_LICENSE("GPL");
-MODULE_IMPORT_NS(PMBUS);
+MODULE_IMPORT_NS("PMBUS");
