Index: sonic-platform-modules-cel/ds2000/modules/Makefile
===================================================================
--- sonic-platform-modules-cel.orig/ds2000/modules/Makefile
+++ sonic-platform-modules-cel/ds2000/modules/Makefile
@@ -1,5 +1,5 @@
 TARGET:= pddf_custom_psu_driver_module
-$(TARGET)-objs := ./psu_driver/pddf_psu_api.o ./psu_driver/pddf_psu_driver.o
+$(TARGET)-objs := psu_driver/pddf_psu_api.o psu_driver/pddf_psu_driver.o
 
 obj-m := $(TARGET).o
 obj-m += mc24lc64t.o pddf_custom_fpga_algo.o pddf_custom_fpga_extend.o lpc_basecpld.o
Index: sonic-platform-modules-cel/ds3000/modules/Makefile
===================================================================
--- sonic-platform-modules-cel.orig/ds3000/modules/Makefile
+++ sonic-platform-modules-cel/ds3000/modules/Makefile
@@ -1,8 +1,8 @@
 LED_TARGET:= pddf_custom_led_module
-$(LED_TARGET)-objs := ./led_driver/pddf_custom_led_module.o
+$(LED_TARGET)-objs := led_driver/pddf_custom_led_module.o
 
 PSU_TARGET:= pddf_custom_psu_driver_module
-$(PSU_TARGET)-objs := ./psu_driver/pddf_psu_api.o ./psu_driver/pddf_psu_driver.o
+$(PSU_TARGET)-objs := psu_driver/pddf_psu_api.o psu_driver/pddf_psu_driver.o
 
 obj-m := mc24lc64t.o baseboard_cpld.o pddf_custom_fpga_algo.o pddf_custom_fpga_extend.o
 obj-m += $(LED_TARGET).o $(PSU_TARGET).o
Index: sonic-platform-modules-cel/silverstone-v2/modules/Makefile
===================================================================
--- sonic-platform-modules-cel.orig/silverstone-v2/modules/Makefile
+++ sonic-platform-modules-cel/silverstone-v2/modules/Makefile
@@ -1,11 +1,11 @@
 TARGET:= pddf_custom_fan_driver_module
-$(TARGET)-objs := ./fan_driver/pddf_custom_fan_api.o ./fan_driver/pddf_custom_fan_driver.o
+$(TARGET)-objs := fan_driver/pddf_custom_fan_api.o fan_driver/pddf_custom_fan_driver.o
 
 TARGET1:= pddf_custom_led_module
-$(TARGET1)-objs := ./led_driver/pddf_custom_led_module.o
+$(TARGET1)-objs := led_driver/pddf_custom_led_module.o
 
 TARGET2:= pddf_custom_psu_driver_module
-$(TARGET2)-objs := ./psu_driver/pddf_psu_api.o ./psu_driver/pddf_psu_driver.o 
+$(TARGET2)-objs := psu_driver/pddf_psu_api.o psu_driver/pddf_psu_driver.o
 
 obj-m := $(TARGET).o $(TARGET1).o $(TARGET2).o
 obj-m += pddf_custom_fpga_algo.o pddf_custom_fpga_extend.o tps536c7.o pddf_custom_wdt.o pddf_custom_lpc_basecpld.o
