Index: sonic-platform-modules-inventec/d6332/modules/gpio-ich.c
===================================================================
--- sonic-platform-modules-inventec.orig/d6332/modules/gpio-ich.c
+++ sonic-platform-modules-inventec/d6332/modules/gpio-ich.c
@@ -274,9 +274,10 @@ static int ich6_gpio_request(struct gpio
 	return ichx_gpio_request(chip, nr);
 }
 
-static void ichx_gpio_set(struct gpio_chip *chip, unsigned nr, int val)
+static int ichx_gpio_set(struct gpio_chip *chip, unsigned nr, int val)
 {
 	ichx_write_bit(GPIO_LVL, nr, val, 0);
+	return 0;
 }
 
 static void ichx_gpiolib_setup(struct gpio_chip *chip)
Index: sonic-platform-modules-inventec/d6356/modules/gpio-ich.c
===================================================================
--- sonic-platform-modules-inventec.orig/d6356/modules/gpio-ich.c
+++ sonic-platform-modules-inventec/d6356/modules/gpio-ich.c
@@ -274,9 +274,10 @@ static int ich6_gpio_request(struct gpio
 	return ichx_gpio_request(chip, nr);
 }
 
-static void ichx_gpio_set(struct gpio_chip *chip, unsigned nr, int val)
+static int ichx_gpio_set(struct gpio_chip *chip, unsigned nr, int val)
 {
 	ichx_write_bit(GPIO_LVL, nr, val, 0);
+	return 0;
 }
 
 static void ichx_gpiolib_setup(struct gpio_chip *chip)
