Index: sonic-platform-modules-tencent/common/modules/rg_spi_gpio.c
===================================================================
--- sonic-platform-modules-tencent.orig/common/modules/rg_spi_gpio.c
+++ sonic-platform-modules-tencent/common/modules/rg_spi_gpio.c
@@ -380,7 +380,7 @@
 		return -ENODEV;
 #endif
 
-	master = spi_alloc_master(&pdev->dev, sizeof(*spi_gpio));
+	master = spi_alloc_target(&pdev->dev, sizeof(*spi_gpio));
 	if (!master)
 		return -ENOMEM;
 
