Index: sonic-platform-modules-tencent/common/modules/rg_plat_sysfs/rg_dev_sysfs/rg_plat_sff.c
===================================================================
--- sonic-platform-modules-tencent.orig/common/modules/rg_plat_sysfs/rg_dev_sysfs/rg_plat_sff.c
+++ sonic-platform-modules-tencent/common/modules/rg_plat_sysfs/rg_dev_sysfs/rg_plat_sff.c
@@ -60,7 +60,7 @@ static ssize_t sff_cpld_info_show(struct
     return ret;
 }
 
-static ssize_t sff_polling_read(struct file *filp, struct kobject *kobj, struct bin_attribute *attr,
+static ssize_t sff_polling_read(struct file *filp, struct kobject *kobj, const struct bin_attribute *attr,
                    char *buf, loff_t offset, size_t count)
 {
     struct switch_obj *sff_obj;
Index: sonic-platform-modules-tencent/common_custom/common_tx/S3IP_sysfs/sysfs_driver/switch.c
===================================================================
--- sonic-platform-modules-tencent.orig/common_custom/common_tx/S3IP_sysfs/sysfs_driver/switch.c
+++ sonic-platform-modules-tencent/common_custom/common_tx/S3IP_sysfs/sysfs_driver/switch.c
@@ -93,7 +93,7 @@ static struct kobj_type switch_ktype = {
     .default_attrs = NULL,
 };
 
-static ssize_t syseeprom_read(struct file *filp, struct kobject *kobj, struct bin_attribute *attr,
+static ssize_t syseeprom_read(struct file *filp, struct kobject *kobj, const struct bin_attribute *attr,
                    char *buf, loff_t offset, size_t count)
 {
     ssize_t rd_len;
@@ -113,7 +113,7 @@ static ssize_t syseeprom_read(struct fil
     return rd_len;
 }
 
-static ssize_t syseeprom_write(struct file *filp, struct kobject *kobj, struct bin_attribute *attr,
+static ssize_t syseeprom_write(struct file *filp, struct kobject *kobj, const struct bin_attribute *attr,
                    char *buf, loff_t offset, size_t count)
 {
     ssize_t wr_len;
Index: sonic-platform-modules-tencent/common_custom/common_tx/S3IP_sysfs/sysfs_driver/transceiver_sysfs.c
===================================================================
--- sonic-platform-modules-tencent.orig/common_custom/common_tx/S3IP_sysfs/sysfs_driver/transceiver_sysfs.c
+++ sonic-platform-modules-tencent/common_custom/common_tx/S3IP_sysfs/sysfs_driver/transceiver_sysfs.c
@@ -306,7 +306,7 @@ static ssize_t eth_interrupt_show(struct
     return ret;
 }
 
-static ssize_t eth_eeprom_read(struct file *filp, struct kobject *kobj, struct bin_attribute *attr,
+static ssize_t eth_eeprom_read(struct file *filp, struct kobject *kobj, const struct bin_attribute *attr,
                    char *buf, loff_t offset, size_t count)
 {
     struct switch_obj *eth_obj;
@@ -332,7 +332,7 @@ static ssize_t eth_eeprom_read(struct fi
     return rd_len;
 }
 
-static ssize_t eth_eeprom_write(struct file *filp, struct kobject *kobj, struct bin_attribute *attr,
+static ssize_t eth_eeprom_write(struct file *filp, struct kobject *kobj, const struct bin_attribute *attr,
                    char *buf, loff_t offset, size_t count)
 {
     struct switch_obj *eth_obj;
