Index: sonic-platform-modules-juniper/common/modules/gpio-tmc.c
===================================================================
--- sonic-platform-modules-juniper.orig/common/modules/gpio-tmc.c
+++ sonic-platform-modules-juniper/common/modules/gpio-tmc.c
@@ -218,7 +218,7 @@ static const u32 sfp_slaveb1_group_offse
 
 struct tmc_gpio_info {
 	int (*get)(struct gpio_chip *, unsigned int);
-	void (*set)(struct gpio_chip *, unsigned int, int);
+	int (*set)(struct gpio_chip *, unsigned int, int);
 	int (*dirin)(struct gpio_chip *, unsigned int);
 	int (*dirout)(struct gpio_chip *, unsigned int, int);
 };
@@ -522,14 +522,14 @@ static int tmc_gpio_get(struct gpio_chip
 /*
  * tmc_gpio_set - Write the specified signal of the GPIO device.
  */
-static void tmc_gpio_set(struct gpio_chip *gc, unsigned int gpio, int val)
+static int tmc_gpio_set(struct gpio_chip *gc, unsigned int gpio, int val)
 {
 	struct tmc_gpio_chip *chip = to_tmc_chip(gc);
 	unsigned int group = TMC_GPIO_FIND_GROUP(gpio);
 	unsigned int bit   = TMC_GPIO_FIND_GPIO(gpio);
 
 	if (group >= TMC_PFE_GPIO_GROUP_MAX)
-		return;
+		return 0;
 
 	switch (group) {
 	case QSFP_RST:
@@ -543,6 +543,8 @@ static void tmc_gpio_set(struct gpio_chi
 		tmc_gpio_multiple_bitsop(chip, bit, group, group_offset[group], val);
 		break;
 	}
+
+	return 0;
 }
 
 static struct tmc_gpio_info tmc_gpios[] = {
