Index: sonic-platform-modules-alphanetworks/bes2348t/modules/bes2348t_onie_eeprom.c
===================================================================
--- sonic-platform-modules-alphanetworks.orig/bes2348t/modules/bes2348t_onie_eeprom.c
+++ sonic-platform-modules-alphanetworks/bes2348t/modules/bes2348t_onie_eeprom.c
@@ -127,8 +127,7 @@ static ssize_t onie_write(struct device
     return count;
 }
 
-static int onie_eeprom_probe(struct i2c_client *client,
-                                 const struct i2c_device_id *dev_id)
+static int onie_eeprom_probe(struct i2c_client *client)
 {
     struct bes2348t_onie_eeprom_data *data;
     int status;
@@ -163,13 +162,12 @@ exit:
     return status;
 }
 
-static int onie_eeprom_remove(struct i2c_client *client)
+static void onie_eeprom_remove(struct i2c_client *client)
 {
     struct bes2348t_onie_eeprom_data *data = i2c_get_clientdata(client);
     sysfs_remove_group(&client->dev.kobj, &bes2348t_onie_group);
     kfree(data);
 
-    return 0;
 }
 
 static const struct i2c_device_id onie_eeprom_id[] = {
Index: sonic-platform-modules-alphanetworks/snh60a0-320fv2/modules/alphanetworks_snh60a0-320fv2_sfp.c
===================================================================
--- sonic-platform-modules-alphanetworks.orig/snh60a0-320fv2/modules/alphanetworks_snh60a0-320fv2_sfp.c
+++ sonic-platform-modules-alphanetworks/snh60a0-320fv2/modules/alphanetworks_snh60a0-320fv2_sfp.c
@@ -1271,7 +1271,7 @@ static ssize_t sfp_port_write(struct sfp
 
 
 static ssize_t sfp_bin_write(struct file *filp, struct kobject *kobj,
-				struct bin_attribute *attr,
+				const struct bin_attribute *attr,
 				char *buf, loff_t off, size_t count)
 {
 	struct sfp_port_data *data;
@@ -1378,7 +1378,7 @@ static ssize_t sfp_port_read(struct sfp_
 }
 
 static ssize_t sfp_bin_read(struct file *filp, struct kobject *kobj,
-		struct bin_attribute *attr,
+		const struct bin_attribute *attr,
 		char *buf, loff_t off, size_t count)
 {
 	struct sfp_port_data *data;
@@ -1521,7 +1521,7 @@ static const struct attribute_group qsfp
 	.attrs = qsfp_attributes,
 };
 
-static int qsfp_probe(struct i2c_client *client, const struct i2c_device_id *dev_id,
+static int qsfp_probe(struct i2c_client *client,
 						  struct qsfp_data **data)
 {
 	int status;
@@ -1570,8 +1570,7 @@ exit:
 	return status;	
 }
 
-static int sfp_device_probe(struct i2c_client *client,
-			const struct i2c_device_id *dev_id)
+static int sfp_device_probe(struct i2c_client *client)
 {
 	struct sfp_port_data *data = NULL;
 
@@ -1582,7 +1581,7 @@ static int sfp_device_probe(struct i2c_c
 
 	i2c_set_clientdata(client, data);
 	mutex_init(&data->update_lock);
-	data->port 	 = dev_id->driver_data;
+	data->port = (uintptr_t)i2c_get_match_data(client);
 	data->client = client;
 
 	if ((client->addr != SFP_EEPROM_A0_I2C_ADDR) && (client->addr != SFP_CPLD_I2C_ADDR)){
@@ -1591,7 +1590,7 @@ static int sfp_device_probe(struct i2c_c
 	}
 	
 	data->driver_type = DRIVER_TYPE_QSFP;
-	return qsfp_probe(client, dev_id, &data->qsfp);
+	return qsfp_probe(client, &data->qsfp);
 }
 
 static int sfp_msa_remove(struct i2c_client *client, struct sfp_msa_data *data)
@@ -1624,20 +1623,21 @@ static int qfp_remove(struct i2c_client
 	return 0;
 }
 
-static int sfp_device_remove(struct i2c_client *client)
+static void sfp_device_remove(struct i2c_client *client)
 {
 	struct sfp_port_data *data = i2c_get_clientdata(client);
 
 	switch (data->driver_type) {
 		case DRIVER_TYPE_SFP_MSA:
-			return sfp_msa_remove(client, data->msa);
+			sfp_msa_remove(client, data->msa);
+			break;
 		case DRIVER_TYPE_SFP_DDM:
-			return sfp_ddm_remove(client, data->ddm);
+			sfp_ddm_remove(client, data->ddm);
+			break;
 		case DRIVER_TYPE_QSFP:
-			return qfp_remove(client, data->qsfp);
+			qfp_remove(client, data->qsfp);
+			break;
    	}
-
-	return 0;
 }
 
 static struct i2c_driver sfp_driver = {
Index: sonic-platform-modules-alphanetworks/snh60a0-320fv2/modules/snh60a0_onie_eeprom.c
===================================================================
--- sonic-platform-modules-alphanetworks.orig/snh60a0-320fv2/modules/snh60a0_onie_eeprom.c
+++ sonic-platform-modules-alphanetworks/snh60a0-320fv2/modules/snh60a0_onie_eeprom.c
@@ -196,8 +196,7 @@ static void alpha_i2c_cpld_remove_client
     mutex_unlock(&list_lock);
 }
 
-static int onie_eeprom_probe(struct i2c_client *client,
-                                 const struct i2c_device_id *dev_id)
+static int onie_eeprom_probe(struct i2c_client *client)
 {
     int status;
 
@@ -221,12 +220,11 @@ exit:
     return status;
 }
 
-static int onie_eeprom_remove(struct i2c_client *client)
+static void onie_eeprom_remove(struct i2c_client *client)
 {
     sysfs_remove_group(&client->dev.kobj, &alphanetworks_snh60a0_320fv2_onie_group);
     alpha_i2c_cpld_remove_client(client);
 
-    return 0;
 }
 
 static const struct i2c_device_id onie_eeprom_id[] = {
Index: sonic-platform-modules-alphanetworks/snh60a0-320fv2/modules/snh60a0_power_cpld.c
===================================================================
--- sonic-platform-modules-alphanetworks.orig/snh60a0-320fv2/modules/snh60a0_power_cpld.c
+++ sonic-platform-modules-alphanetworks/snh60a0-320fv2/modules/snh60a0_power_cpld.c
@@ -356,8 +356,7 @@ static void alpha_i2c_cpld_remove_client
     mutex_unlock(&list_lock);
 }
 
-static int alpha_i2c_cpld_probe(struct i2c_client *client,
-                                 const struct i2c_device_id *dev_id)
+static int alpha_i2c_cpld_probe(struct i2c_client *client)
 {
     int status;
     struct alphanetworks_snh60a0_320fv2_pwr_cpld_data* data;
@@ -397,14 +396,13 @@ exit:
     return status;
 }
 
-static int alpha_i2c_cpld_remove(struct i2c_client *client)
+static void alpha_i2c_cpld_remove(struct i2c_client *client)
 {
     struct alphanetworks_snh60a0_320fv2_pwr_cpld_data *data = i2c_get_clientdata(client);
     sysfs_remove_group(&client->dev.kobj, &alphanetworks_snh60a0_320fv2_psu_group);
     alpha_i2c_cpld_remove_client(client);
     kfree(data);
 
-    return 0;
 }
 
 static const struct i2c_device_id alpha_i2c_cpld_id[] = {
Index: sonic-platform-modules-alphanetworks/snh60a0-320fv2/modules/snh60a0_system_cpld.c
===================================================================
--- sonic-platform-modules-alphanetworks.orig/snh60a0-320fv2/modules/snh60a0_system_cpld.c
+++ sonic-platform-modules-alphanetworks/snh60a0-320fv2/modules/snh60a0_system_cpld.c
@@ -309,8 +309,7 @@ static void alpha_i2c_cpld_remove_client
     mutex_unlock(&list_lock);
 }
 
-static int alpha_i2c_cpld_probe(struct i2c_client *client,
-                                 const struct i2c_device_id *dev_id)
+static int alpha_i2c_cpld_probe(struct i2c_client *client)
 {
     int status;
 
@@ -334,12 +333,11 @@ exit:
     return status;
 }
 
-static int alpha_i2c_cpld_remove(struct i2c_client *client)
+static void alpha_i2c_cpld_remove(struct i2c_client *client)
 {
     sysfs_remove_group(&client->dev.kobj, &alphanetworks_snh60a0_320fv2_sys_group);
     alpha_i2c_cpld_remove_client(client);
 
-    return 0;
 }
 
 static const struct i2c_device_id alpha_i2c_cpld_id[] = {
Index: sonic-platform-modules-alphanetworks/snh60b0-640f/modules/alphanetworks_snh60b0-640f_sfp.c
===================================================================
--- sonic-platform-modules-alphanetworks.orig/snh60b0-640f/modules/alphanetworks_snh60b0-640f_sfp.c
+++ sonic-platform-modules-alphanetworks/snh60b0-640f/modules/alphanetworks_snh60b0-640f_sfp.c
@@ -1669,7 +1669,7 @@ static ssize_t sfp_port_write(struct sfp
 
 
 static ssize_t sfp_bin_write(struct file *filp, struct kobject *kobj,
-				struct bin_attribute *attr,
+				const struct bin_attribute *attr,
 				char *buf, loff_t off, size_t count)
 {
 	struct sfp_port_data *data;
@@ -1776,7 +1776,7 @@ static ssize_t sfp_port_read(struct sfp_
 }
 
 static ssize_t sfp_bin_read(struct file *filp, struct kobject *kobj,
-		struct bin_attribute *attr,
+		const struct bin_attribute *attr,
 		char *buf, loff_t off, size_t count)
 {
 	struct sfp_port_data *data;
@@ -1919,7 +1919,7 @@ static const struct attribute_group qsfp
 	.attrs = qsfp_attributes,
 };
 
-static int qsfp_probe(struct i2c_client *client, const struct i2c_device_id *dev_id,
+static int qsfp_probe(struct i2c_client *client,
 						  struct qsfp_data **data)
 {
 	int status;
@@ -1968,8 +1968,7 @@ exit:
 	return status;	
 }
 
-static int sfp_device_probe(struct i2c_client *client,
-			const struct i2c_device_id *dev_id)
+static int sfp_device_probe(struct i2c_client *client)
 {
 	struct sfp_port_data *data = NULL;
 
@@ -1980,7 +1979,7 @@ static int sfp_device_probe(struct i2c_c
 
 	i2c_set_clientdata(client, data);
 	mutex_init(&data->update_lock);
-	data->port 	 = dev_id->driver_data;
+	data->port = (uintptr_t)i2c_get_match_data(client);
 	data->client = client;
 
 	DEBUG_PRINT("data->port:%d client->addr:0x%0x\n", data->port, client->addr);
@@ -1998,7 +1997,7 @@ static int sfp_device_probe(struct i2c_c
 	}
 	
 	data->driver_type = DRIVER_TYPE_QSFP;
-	return qsfp_probe(client, dev_id, &data->qsfp);
+	return qsfp_probe(client, &data->qsfp);
 }
 
 static int sfp_msa_remove(struct i2c_client *client, struct sfp_msa_data *data)
@@ -2031,20 +2030,21 @@ static int qfp_remove(struct i2c_client
 	return 0;
 }
 
-static int sfp_device_remove(struct i2c_client *client)
+static void sfp_device_remove(struct i2c_client *client)
 {
 	struct sfp_port_data *data = i2c_get_clientdata(client);
 
 	switch (data->driver_type) {
 		case DRIVER_TYPE_SFP_MSA:
-			return sfp_msa_remove(client, data->msa);
+			sfp_msa_remove(client, data->msa);
+			break;
 		case DRIVER_TYPE_SFP_DDM:
-			return sfp_ddm_remove(client, data->ddm);
+			sfp_ddm_remove(client, data->ddm);
+			break;
 		case DRIVER_TYPE_QSFP:
-			return qfp_remove(client, data->qsfp);
+			qfp_remove(client, data->qsfp);
+			break;
    	}
-
-	return 0;
 }
 
 static struct i2c_driver sfp_driver = {
Index: sonic-platform-modules-alphanetworks/snh60b0-640f/modules/snh60b0-640f_onie_eeprom.c
===================================================================
--- sonic-platform-modules-alphanetworks.orig/snh60b0-640f/modules/snh60b0-640f_onie_eeprom.c
+++ sonic-platform-modules-alphanetworks/snh60b0-640f/modules/snh60b0-640f_onie_eeprom.c
@@ -196,8 +196,7 @@ static void alpha_i2c_cpld_remove_client
     mutex_unlock(&list_lock);
 }
 
-static int onie_eeprom_probe(struct i2c_client *client,
-                                 const struct i2c_device_id *dev_id)
+static int onie_eeprom_probe(struct i2c_client *client)
 {
     int status;
 
@@ -221,12 +220,11 @@ exit:
     return status;
 }
 
-static int onie_eeprom_remove(struct i2c_client *client)
+static void onie_eeprom_remove(struct i2c_client *client)
 {
     sysfs_remove_group(&client->dev.kobj, &snh60b0_onie_group);
     alpha_i2c_cpld_remove_client(client);
 
-    return 0;
 }
 
 static const struct i2c_device_id onie_eeprom_id[] = {
Index: sonic-platform-modules-alphanetworks/snh60b0-640f/modules/snh60b0-640f_power_cpld.c
===================================================================
--- sonic-platform-modules-alphanetworks.orig/snh60b0-640f/modules/snh60b0-640f_power_cpld.c
+++ sonic-platform-modules-alphanetworks/snh60b0-640f/modules/snh60b0-640f_power_cpld.c
@@ -350,8 +350,7 @@ static void alpha_i2c_cpld_remove_client
     mutex_unlock(&list_lock);
 }
 
-static int alpha_i2c_cpld_probe(struct i2c_client *client,
-                                 const struct i2c_device_id *dev_id)
+static int alpha_i2c_cpld_probe(struct i2c_client *client)
 {
     int status;
     struct snh60b0_pwr_cpld_data* data;
@@ -391,14 +390,13 @@ exit:
     return status;
 }
 
-static int alpha_i2c_cpld_remove(struct i2c_client *client)
+static void alpha_i2c_cpld_remove(struct i2c_client *client)
 {
     struct snh60b0_pwr_cpld_data *data = i2c_get_clientdata(client);
     sysfs_remove_group(&client->dev.kobj, &snh60b0_psu_group);
     alpha_i2c_cpld_remove_client(client);
     kfree(data);
 
-    return 0;
 }
 
 static const struct i2c_device_id alpha_i2c_cpld_id[] = {
Index: sonic-platform-modules-alphanetworks/snh60b0-640f/modules/snh60b0-640f_system_cpld.c
===================================================================
--- sonic-platform-modules-alphanetworks.orig/snh60b0-640f/modules/snh60b0-640f_system_cpld.c
+++ sonic-platform-modules-alphanetworks/snh60b0-640f/modules/snh60b0-640f_system_cpld.c
@@ -294,8 +294,7 @@ static void alpha_i2c_cpld_remove_client
     mutex_unlock(&list_lock);
 }
 
-static int alpha_i2c_cpld_probe(struct i2c_client *client,
-                                 const struct i2c_device_id *dev_id)
+static int alpha_i2c_cpld_probe(struct i2c_client *client)
 {
     int status;
 
@@ -319,12 +318,11 @@ exit:
     return status;
 }
 
-static int alpha_i2c_cpld_remove(struct i2c_client *client)
+static void alpha_i2c_cpld_remove(struct i2c_client *client)
 {
     sysfs_remove_group(&client->dev.kobj, &snh60b0_sys_group);
     alpha_i2c_cpld_remove_client(client);
 
-    return 0;
 }
 
 static const struct i2c_device_id alpha_i2c_cpld_id[] = {
Index: sonic-platform-modules-alphanetworks/snj60d0-320f/modules/snj60d0-320f_fpga.c
===================================================================
--- sonic-platform-modules-alphanetworks.orig/snj60d0-320f/modules/snj60d0-320f_fpga.c
+++ sonic-platform-modules-alphanetworks/snj60d0-320f/modules/snj60d0-320f_fpga.c
@@ -808,8 +808,7 @@ static void alpha_i2c_fpga_remove_client
     mutex_unlock(&list_lock);
 }
 
-static int alpha_i2c_fpga_probe(struct i2c_client *client,
-                                 const struct i2c_device_id *dev_id)
+static int alpha_i2c_fpga_probe(struct i2c_client *client)
 {
     int status;
     struct alphanetworks_snj60d0_320f_pwr_fpga_data* data;
@@ -849,14 +848,13 @@ exit:
     return status;
 }
 
-static int alpha_i2c_fpga_remove(struct i2c_client *client)
+static void alpha_i2c_fpga_remove(struct i2c_client *client)
 {
     struct alphanetworks_snj60d0_320f_pwr_fpga_data *data = i2c_get_clientdata(client);
     sysfs_remove_group(&client->dev.kobj, &alphanetworks_snj60d0_320f_fpga_group);
     alpha_i2c_fpga_remove_client(client);
     kfree(data);
 
-    return 0;
 }
 
 static const struct i2c_device_id alpha_i2c_fpga_id[] = {
Index: sonic-platform-modules-alphanetworks/snj60d0-320f/modules/snj60d0-320f_i2c_mux_cpld.c
===================================================================
--- sonic-platform-modules-alphanetworks.orig/snj60d0-320f/modules/snj60d0-320f_i2c_mux_cpld.c
+++ sonic-platform-modules-alphanetworks/snj60d0-320f/modules/snj60d0-320f_i2c_mux_cpld.c
@@ -1063,11 +1063,10 @@ static ssize_t show_version(struct devic
 /*
  * I2C init/probing/exit functions
  */
-static int snj60d0_320f_cpld_mux_probe(struct i2c_client *client,
-                                     const struct i2c_device_id *id)
+static int snj60d0_320f_cpld_mux_probe(struct i2c_client *client)
 {
     struct i2c_adapter *adap = to_i2c_adapter(client->dev.parent);
-    int num, force, class;
+    int num, force;
     struct i2c_mux_core *muxc;
     struct snj60d0_320f_cpld_data *data;
     int ret = 0;
@@ -1077,7 +1076,7 @@ static int snj60d0_320f_cpld_mux_probe(s
         return -ENODEV;
 
     muxc = i2c_mux_alloc(adap, &client->dev,
-                         chips[id->driver_data].nchans, sizeof(*data), 0,
+                         chips[(uintptr_t)i2c_get_match_data(client)].nchans, sizeof(*data), 0,
                          snj60d0_320f_cpld_mux_select_chan, snj60d0_320f_cpld_mux_deselect_mux);
     if (!muxc)
         return -ENOMEM;
@@ -1085,16 +1084,15 @@ static int snj60d0_320f_cpld_mux_probe(s
     i2c_set_clientdata(client, muxc);
     data = i2c_mux_priv(muxc);
     data->client = client;
-    data->type = id->driver_data;
+    data->type = (uintptr_t)i2c_get_match_data(client);
     data->last_chan = chips[data->type].deselectChan;    /* force the first selection */
     mutex_init(&data->update_lock);
 
     /* Now create an adapter for each channel */
     for (num = 0; num < chips[data->type].nchans; num++) {
         force = 0;              /* dynamic adap number */
-        class = 0;              /* no class by default */
 
-        ret = i2c_mux_add_adapter(muxc, force, num, class);
+        ret = i2c_mux_add_adapter(muxc, force, num);
 
         if (ret) {
             dev_err(&client->dev,
@@ -1159,7 +1157,7 @@ add_mux_failed:
     return ret;
 }
 
-static int snj60d0_320f_cpld_mux_remove(struct i2c_client *client)
+static void snj60d0_320f_cpld_mux_remove(struct i2c_client *client)
 {
     struct i2c_mux_core *muxc = i2c_get_clientdata(client);
     struct snj60d0_320f_cpld_data *data = i2c_mux_priv(muxc);
@@ -1190,8 +1188,6 @@ static int snj60d0_320f_cpld_mux_remove(
     }
 
     i2c_mux_del_adapters(muxc);
-
-    return 0;
 }
 
 static int snj60d0_320f_cpld_read_internal(struct i2c_client *client, u8 reg)
Index: sonic-platform-modules-alphanetworks/snj60d0-320f/modules/snj60d0-320f_onie_eeprom.c
===================================================================
--- sonic-platform-modules-alphanetworks.orig/snj60d0-320f/modules/snj60d0-320f_onie_eeprom.c
+++ sonic-platform-modules-alphanetworks/snj60d0-320f/modules/snj60d0-320f_onie_eeprom.c
@@ -127,8 +127,7 @@ static ssize_t onie_write(struct device
     return count;
 }
 
-static int onie_eeprom_probe(struct i2c_client *client,
-                                 const struct i2c_device_id *dev_id)
+static int onie_eeprom_probe(struct i2c_client *client)
 {
     struct snj60d0_320f_onie_eeprom_data *data;
     int status;
@@ -163,13 +162,12 @@ exit:
     return status;
 }
 
-static int onie_eeprom_remove(struct i2c_client *client)
+static void onie_eeprom_remove(struct i2c_client *client)
 {
     struct snj60d0_320f_onie_eeprom_data *data = i2c_get_clientdata(client);
     sysfs_remove_group(&client->dev.kobj, &snj60d0_onie_group);
     kfree(data);
 
-    return 0;
 }
 
 static const struct i2c_device_id onie_eeprom_id[] = {
Index: sonic-platform-modules-alphanetworks/snj60d0-320f/modules/snj60d0-320f_sfp.c
===================================================================
--- sonic-platform-modules-alphanetworks.orig/snj60d0-320f/modules/snj60d0-320f_sfp.c
+++ sonic-platform-modules-alphanetworks/snj60d0-320f/modules/snj60d0-320f_sfp.c
@@ -1038,7 +1038,7 @@ static ssize_t sfp_port_write(struct sfp
 }
 
 static ssize_t sfp_bin_write(struct file *filp, struct kobject *kobj,
-				struct bin_attribute *attr,
+				const struct bin_attribute *attr,
 				char *buf, loff_t off, size_t count)
 {
 	struct sfp_port_data *data;
@@ -1145,7 +1145,7 @@ static ssize_t sfp_port_read(struct sfp_
 }
 
 static ssize_t sfp_bin_read(struct file *filp, struct kobject *kobj,
-		struct bin_attribute *attr,
+		const struct bin_attribute *attr,
 		char *buf, loff_t off, size_t count)
 {
 	struct sfp_port_data *data;
@@ -1193,7 +1193,7 @@ static const struct attribute_group qsfp
 	.attrs = qsfp_attributes,
 };
 
-static int qsfp_probe(struct i2c_client *client, const struct i2c_device_id *dev_id,
+static int qsfp_probe(struct i2c_client *client,
 						  struct qsfp_data **data)
 {
     int status;
@@ -1246,8 +1246,7 @@ exit:
     return status;	
 }
 
-static int qsfp_device_probe(struct i2c_client *client,
-			const struct i2c_device_id *dev_id)
+static int qsfp_device_probe(struct i2c_client *client)
 {
     struct sfp_port_data *data = NULL;
 
@@ -1258,7 +1257,7 @@ static int qsfp_device_probe(struct i2c_
 
     i2c_set_clientdata(client, data);
     mutex_init(&data->update_lock);
-    data->port 	 = dev_id->driver_data;
+    data->port = (uintptr_t)i2c_get_match_data(client);
     data->client = client;
 
     if (client->addr == SFP_CPLD_I2C_ADDR){
@@ -1272,7 +1271,7 @@ static int qsfp_device_probe(struct i2c_
     if(data->port < SFPPLUS_1_PORT_NUMBER){
 		/* QSFP port probe */
     	data->driver_type = DRIVER_TYPE_QSFP;
-    	return qsfp_probe(client, dev_id, &data->qsfp);
+    	return qsfp_probe(client, &data->qsfp);
 	}
 
 	return 0;
@@ -1288,14 +1287,12 @@ static int qsfp_remove(struct i2c_client
     return 0;
 }
 
-static int qsfp_device_remove(struct i2c_client *client)
+static void qsfp_device_remove(struct i2c_client *client)
 {
     struct sfp_port_data *data = i2c_get_clientdata(client);
 
 	if (data->driver_type == DRIVER_TYPE_QSFP)
-		return qsfp_remove(client, data->qsfp);
-
-    return 0;
+		qsfp_remove(client, data->qsfp);
 }
 
 static struct i2c_driver qsfp_driver = {
Index: sonic-platform-modules-alphanetworks/snj60d0-320f/modules/yesm1300am.c
===================================================================
--- sonic-platform-modules-alphanetworks.orig/snj60d0-320f/modules/yesm1300am.c
+++ sonic-platform-modules-alphanetworks/snj60d0-320f/modules/yesm1300am.c
@@ -242,8 +242,7 @@ static const struct attribute_group yesm
     .attrs = yesm1300am_attributes,
 };
 
-static int yesm1300am_probe(struct i2c_client *client,
-                         const struct i2c_device_id *dev_id)
+static int yesm1300am_probe(struct i2c_client *client)
 {
     struct yesm1300am_data *data;
     int status;
@@ -296,7 +295,7 @@ exit:
     return status;
 }
 
-static int yesm1300am_remove(struct i2c_client *client)
+static void yesm1300am_remove(struct i2c_client *client)
 {
     struct yesm1300am_data *data = i2c_get_clientdata(client);
 
@@ -304,7 +303,6 @@ static int yesm1300am_remove(struct i2c_
     sysfs_remove_group(&client->dev.kobj, &yesm1300am_group);
     kfree(data);
 
-    return 0;
 }
 
 static const struct i2c_device_id yesm1300am_id[] = {
