Index: sonic-platform-modules-ragile/common/modules/wb_optoe.c
===================================================================
--- sonic-platform-modules-ragile.orig/common/modules/wb_optoe.c
+++ sonic-platform-modules-ragile/common/modules/wb_optoe.c
@@ -771,7 +771,7 @@ static ssize_t optoe_read_write(struct o
 }
 
 static ssize_t optoe_bin_read(struct file *filp, struct kobject *kobj,
-		struct bin_attribute *attr,
+		const struct bin_attribute *attr,
 		char *buf, loff_t off, size_t count)
 {
 	struct i2c_client *client = to_i2c_client(container_of(kobj,
@@ -782,7 +782,7 @@ static ssize_t optoe_bin_read(struct fil
 }
 
 static ssize_t optoe_bin_write(struct file *filp, struct kobject *kobj,
-		struct bin_attribute *attr,
+		const struct bin_attribute *attr,
 		char *buf, loff_t off, size_t count)
 {
 	struct i2c_client *client = to_i2c_client(container_of(kobj,
