Index: sonic-platform-modules-ragile/common/modules/linux-5.10/wb_i2c_i801.c
===================================================================
--- sonic-platform-modules-ragile.orig/common/modules/linux-5.10/wb_i2c_i801.c
+++ sonic-platform-modules-ragile/common/modules/linux-5.10/wb_i2c_i801.c
@@ -88,6 +88,7 @@
 
 #include <linux/interrupt.h>
 #include <linux/module.h>
+#include <linux/version.h>
 #include <linux/pci.h>
 #include <linux/kernel.h>
 #include <linux/stddef.h>
@@ -1429,10 +1430,12 @@ static void i801_probe_optional_slaves(s
 		register_dell_lis3lv02d_i2c_device(priv);
 
 	/* Instantiate SPD EEPROMs unless the SMBus is multiplexed */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 11, 0)
 #if IS_ENABLED(CONFIG_I2C_MUX_GPIO)
 	if (!priv->mux_drvdata)
 #endif
 		i2c_register_spd(&priv->adapter);
+#endif
 }
 #else
 static void __init input_apanel_init(void) {}
