Index: sonic-platform-modules-delta/debian/control
===================================================================
--- sonic-platform-modules-delta.orig/debian/control
+++ sonic-platform-modules-delta/debian/control
@@ -7,30 +7,30 @@ Standards-Version: 3.9.3
 
 Package: platform-modules-ag9032v1
 Architecture: amd64
-Depends: linux-image-4.19.0-12-2-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
 
 Package: platform-modules-ag9064
 Architecture: amd64
-Depends: linux-image-4.19.0-12-2-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
 
 Package: platform-modules-ag5648
 Architecture: amd64
-Depends: linux-image-4.19.0-12-2-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
 
 Package: platform-modules-et-6248brb
 Architecture: amd64
-Depends: linux-image-4.19.0-12-2-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
 
 Package: platform-modules-ag9032v2a
 Architecture: amd64
-Depends: linux-image-4.19.0-12-2-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as syseeprom, sfp
 
 Package: platform-modules-agc032
 Architecture: amd64
-Depends: linux-image-4.19.0-12-2-amd64-unsigned
+Depends: linux-image-7.0.0-1002-sonic
 Description: kernel modules for platform devices such as fan, led, sfp
Index: sonic-platform-modules-delta/et-6248brb/modules/dni_gpio.c
===================================================================
--- sonic-platform-modules-delta.orig/et-6248brb/modules/dni_gpio.c
+++ sonic-platform-modules-delta/et-6248brb/modules/dni_gpio.c
@@ -286,9 +286,10 @@ static int ich6_gpio_request(struct gpio_chip *chip, unsigned nr)
 	return ichx_gpio_request(chip, nr);
 }
 
-static void ichx_gpio_set(struct gpio_chip *chip, unsigned nr, int val)
+static int ichx_gpio_set(struct gpio_chip *chip, unsigned nr, int val)
 {
 	ichx_write_bit(GPIO_LVL, nr, val, 0);
+	return 0;
 }
 
 static void ichx_gpiolib_setup(struct gpio_chip *chip)
