Index: sonic-platform-modules-arista/src/scd-spi.c
===================================================================
--- sonic-platform-modules-arista.orig/src/scd-spi.c
+++ sonic-platform-modules-arista/src/scd-spi.c
@@ -249,7 +249,7 @@
    dev_dbg(dev, "adding spi controller %d at addr %#x\n",
            bus, addr);
 
-   controller = spi_alloc_master(dev, sizeof(struct scd_spi_controller));
+   controller = spi_alloc_target(dev, sizeof(struct scd_spi_controller));
    if (!controller) {
       return -ENOMEM;
    }
