Index: sonic-platform-modules-arista/src/scd-uart.c
===================================================================
--- sonic-platform-modules-arista.orig/src/scd-uart.c
+++ sonic-platform-modules-arista/src/scd-uart.c
@@ -301,7 +301,7 @@ static enum hrtimer_restart scd_uart_por
       spin_unlock(&sp->port.lock);
    }
    scd_uart_receive_chars(&sp->port);
-   hrtimer_forward(timer, timer->base->get_time(), scd_uart_poll_interval(sp));
+   hrtimer_forward(timer, hrtimer_cb_get_time(timer), scd_uart_poll_interval(sp));
    return HRTIMER_RESTART;
 }
 
@@ -318,8 +318,7 @@ static int scd_uart_startup(struct uart_
 
    scd_uart_flush_rx_queue(&sp->port);
 
-   hrtimer_init(&sp->timer, CLOCK_MONOTONIC, HRTIMER_MODE_REL);
-   sp->timer.function = scd_uart_port_timer_callback;
+   hrtimer_setup(&sp->timer, scd_uart_port_timer_callback, CLOCK_MONOTONIC, HRTIMER_MODE_REL);
    hrtimer_start(&sp->timer, sp->poll_interval, HRTIMER_MODE_REL);
 
    return 0;
