Index: sonic-platform-modules-accton/debian/control
===================================================================
--- sonic-platform-modules-accton.orig/debian/control
+++ sonic-platform-modules-accton/debian/control
@@ -1,7 +1,7 @@
 Source: sonic-accton-platform-modules
 Section: main
 Priority: extra
-Maintainer: Accton network <roy_lee@accton.com>, Accton Network <polly_hsu@accton.com>, Accton Network <jostar_yang@accton.com>, Accton Network <brandon_chuang@accton.com>
+Maintainer: Accton network <roy_lee@accton.com>
 Build-Depends: debhelper-compat (= 13), python3
 Rules-Requires-Root: no
 Standards-Version: 3.9.3
