Index: sonic-platform-modules-accton/as7712-32x/modules/accton_as7712_32x_fan.c
===================================================================
--- sonic-platform-modules-accton.orig/as7712-32x/modules/accton_as7712_32x_fan.c
+++ sonic-platform-modules-accton/as7712-32x/modules/accton_as7712_32x_fan.c
@@ -343,7 +343,7 @@ struct hwmon_device {
 #define to_hwmon_device(d) container_of(d, struct hwmon_device, dev)
 
 /*Find the 1st valid dev of all childs, it supposes to have only 1 child.*/
-static int is_valid_dev(struct device *dev, void *data)
+static int is_valid_dev(struct device *dev, const void *data)
 {
     struct hwmon_device *hwmon_dev;
 
