Index: sonic-platform-modules-ufispace/s9300-32d/modules/x86-64-ufispace-s9300-32d-i2c-smbus.c
===================================================================
--- sonic-platform-modules-ufispace.orig/s9300-32d/modules/x86-64-ufispace-s9300-32d-i2c-smbus.c
+++ sonic-platform-modules-ufispace/s9300-32d/modules/x86-64-ufispace-s9300-32d-i2c-smbus.c
@@ -22,8 +22,10 @@
 #include <linux/workqueue.h>
 #include <linux/version.h>
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 11, 0)
 static int disable_i2c_register_spd = 1;
 module_param(disable_i2c_register_spd, int, 0);
+#endif
 
 struct i2c_smbus_alert {
 	struct work_struct	alert;
@@ -385,6 +387,7 @@ EXPORT_SYMBOL_GPL(i2c_free_slave_host_no
  *  - Only works on systems with 1 to 8 memory slots
  */
 #if IS_ENABLED(CONFIG_DMI)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 11, 0)
 void i2c_register_spd(struct i2c_adapter *adap)
 {
 	int n, slot_count = 0, dimm_count = 0;
@@ -489,6 +492,7 @@ void i2c_register_spd(struct i2c_adapter
 }
 EXPORT_SYMBOL_GPL(i2c_register_spd);
 #endif
+#endif
 
 MODULE_AUTHOR("Jean Delvare <jdelvare@suse.de>");
 MODULE_DESCRIPTION("SMBus protocol extensions support");
Index: sonic-platform-modules-ufispace/s9301-32d/modules/x86-64-ufispace-s9301-32d-i2c-smbus.c
===================================================================
--- sonic-platform-modules-ufispace.orig/s9301-32d/modules/x86-64-ufispace-s9301-32d-i2c-smbus.c
+++ sonic-platform-modules-ufispace/s9301-32d/modules/x86-64-ufispace-s9301-32d-i2c-smbus.c
@@ -22,8 +22,10 @@
 #include <linux/workqueue.h>
 #include <linux/version.h>
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 11, 0)
 static int disable_i2c_register_spd = 1;
 module_param(disable_i2c_register_spd, int, 0);
+#endif
 
 struct i2c_smbus_alert {
 	struct work_struct	alert;
@@ -385,6 +387,7 @@ EXPORT_SYMBOL_GPL(i2c_free_slave_host_no
  *  - Only works on systems with 1 to 8 memory slots
  */
 #if IS_ENABLED(CONFIG_DMI)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 11, 0)
 void i2c_register_spd(struct i2c_adapter *adap)
 {
 	int n, slot_count = 0, dimm_count = 0;
@@ -489,6 +492,7 @@ void i2c_register_spd(struct i2c_adapter
 }
 EXPORT_SYMBOL_GPL(i2c_register_spd);
 #endif
+#endif
 
 MODULE_AUTHOR("Jean Delvare <jdelvare@suse.de>");
 MODULE_DESCRIPTION("SMBus protocol extensions support");
Index: sonic-platform-modules-ufispace/s9301-32db/modules/x86-64-ufispace-s9301-32db-i2c-smbus.c
===================================================================
--- sonic-platform-modules-ufispace.orig/s9301-32db/modules/x86-64-ufispace-s9301-32db-i2c-smbus.c
+++ sonic-platform-modules-ufispace/s9301-32db/modules/x86-64-ufispace-s9301-32db-i2c-smbus.c
@@ -22,8 +22,10 @@
 #include <linux/workqueue.h>
 #include <linux/version.h>
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 11, 0)
 static int disable_i2c_register_spd = 1;
 module_param(disable_i2c_register_spd, int, 0);
+#endif
 
 struct i2c_smbus_alert {
 	struct work_struct	alert;
@@ -385,6 +387,7 @@ EXPORT_SYMBOL_GPL(i2c_free_slave_host_no
  *  - Only works on systems with 1 to 8 memory slots
  */
 #if IS_ENABLED(CONFIG_DMI)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 11, 0)
 void i2c_register_spd(struct i2c_adapter *adap)
 {
 	int n, slot_count = 0, dimm_count = 0;
@@ -489,6 +492,7 @@ void i2c_register_spd(struct i2c_adapter
 }
 EXPORT_SYMBOL_GPL(i2c_register_spd);
 #endif
+#endif
 
 MODULE_AUTHOR("Jean Delvare <jdelvare@suse.de>");
 MODULE_DESCRIPTION("SMBus protocol extensions support");
Index: sonic-platform-modules-ufispace/s9321-64e/modules/x86-64-ufispace-s9321-64e-i2c-smbus.c
===================================================================
--- sonic-platform-modules-ufispace.orig/s9321-64e/modules/x86-64-ufispace-s9321-64e-i2c-smbus.c
+++ sonic-platform-modules-ufispace/s9321-64e/modules/x86-64-ufispace-s9321-64e-i2c-smbus.c
@@ -22,8 +22,10 @@
 #include <linux/workqueue.h>
 #include <linux/version.h>
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 11, 0)
 static int disable_i2c_register_spd = 1;
 module_param(disable_i2c_register_spd, int, 0);
+#endif
 
 struct i2c_smbus_alert {
 	struct work_struct	alert;
@@ -385,6 +387,7 @@ EXPORT_SYMBOL_GPL(i2c_free_slave_host_no
  *  - Only works on systems with 1 to 8 memory slots
  */
 #if IS_ENABLED(CONFIG_DMI)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 11, 0)
 void i2c_register_spd(struct i2c_adapter *adap)
 {
 	int n, slot_count = 0, dimm_count = 0;
@@ -489,6 +492,7 @@ void i2c_register_spd(struct i2c_adapter
 }
 EXPORT_SYMBOL_GPL(i2c_register_spd);
 #endif
+#endif
 
 MODULE_AUTHOR("Jean Delvare <jdelvare@suse.de>");
 MODULE_DESCRIPTION("SMBus protocol extensions support");
Index: sonic-platform-modules-ufispace/s9321-64eo/modules/x86-64-ufispace-s9321-64eo-i2c-smbus.c
===================================================================
--- sonic-platform-modules-ufispace.orig/s9321-64eo/modules/x86-64-ufispace-s9321-64eo-i2c-smbus.c
+++ sonic-platform-modules-ufispace/s9321-64eo/modules/x86-64-ufispace-s9321-64eo-i2c-smbus.c
@@ -22,8 +22,10 @@
 #include <linux/workqueue.h>
 #include <linux/version.h>
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 11, 0)
 static int disable_i2c_register_spd = 1;
 module_param(disable_i2c_register_spd, int, 0);
+#endif
 
 struct i2c_smbus_alert {
 	struct work_struct	alert;
@@ -385,6 +387,7 @@ EXPORT_SYMBOL_GPL(i2c_free_slave_host_no
  *  - Only works on systems with 1 to 8 memory slots
  */
 #if IS_ENABLED(CONFIG_DMI)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 11, 0)
 void i2c_register_spd(struct i2c_adapter *adap)
 {
 	int n, slot_count = 0, dimm_count = 0;
@@ -489,6 +492,7 @@ void i2c_register_spd(struct i2c_adapter
 }
 EXPORT_SYMBOL_GPL(i2c_register_spd);
 #endif
+#endif
 
 MODULE_AUTHOR("Jean Delvare <jdelvare@suse.de>");
 MODULE_DESCRIPTION("SMBus protocol extensions support");
