Index: sonic-platform-modules-micas/common/modules/s3ip_sysfs/sysfs_driver/eeprom_sysfs.c
===================================================================
--- sonic-platform-modules-micas.orig/common/modules/s3ip_sysfs/sysfs_driver/eeprom_sysfs.c
+++ sonic-platform-modules-micas/common/modules/s3ip_sysfs/sysfs_driver/eeprom_sysfs.c
@@ -108,7 +108,7 @@ static ssize_t eeprom_type_show(struct s
     return g_eeprom_drv->get_eeprom_type(eeprom_index, buf, PAGE_SIZE);
 }
 
-static ssize_t eeprom_eeprom_read(struct file *filp, struct kobject *kobj, struct bin_attribute *attr,
+static ssize_t eeprom_eeprom_read(struct file *filp, struct kobject *kobj, const struct bin_attribute *attr,
                    char *buf, loff_t offset, size_t count)
 {
     struct switch_obj *eeprom_obj;
@@ -134,7 +134,7 @@ static ssize_t eeprom_eeprom_read(struct
     return rd_len;
 }
 
-static ssize_t eeprom_eeprom_write(struct file *filp, struct kobject *kobj, struct bin_attribute *attr,
+static ssize_t eeprom_eeprom_write(struct file *filp, struct kobject *kobj, const struct bin_attribute *attr,
                    char *buf, loff_t offset, size_t count)
 {
     struct switch_obj *eeprom_obj;
Index: sonic-platform-modules-micas/common/modules/s3ip_sysfs/sysfs_driver/psu_sysfs.c
===================================================================
--- sonic-platform-modules-micas.orig/common/modules/s3ip_sysfs/sysfs_driver/psu_sysfs.c
+++ sonic-platform-modules-micas/common/modules/s3ip_sysfs/sysfs_driver/psu_sysfs.c
@@ -602,7 +602,7 @@ static ssize_t psu_attr_threshold_show(s
     return g_psu_drv->get_psu_attr_threshold(psu_index, tmp_attr->type, buf, PAGE_SIZE);
 }
 
-static ssize_t psu_eeprom_read(struct file *filp, struct kobject *kobj, struct bin_attribute *attr,
+static ssize_t psu_eeprom_read(struct file *filp, struct kobject *kobj, const struct bin_attribute *attr,
                                char *buf, loff_t offset, size_t count)
 {
     struct switch_obj *psu_obj;
Index: sonic-platform-modules-micas/common/modules/s3ip_sysfs/sysfs_driver/switch.c
===================================================================
--- sonic-platform-modules-micas.orig/common/modules/s3ip_sysfs/sysfs_driver/switch.c
+++ sonic-platform-modules-micas/common/modules/s3ip_sysfs/sysfs_driver/switch.c
@@ -100,7 +100,7 @@ static struct kobj_type switch_ktype = {
     .release = switch_obj_release,
 };
 
-static ssize_t syseeprom_read(struct file *filp, struct kobject *kobj, struct bin_attribute *attr,
+static ssize_t syseeprom_read(struct file *filp, struct kobject *kobj, const struct bin_attribute *attr,
                    char *buf, loff_t offset, size_t count)
 {
     ssize_t rd_len;
@@ -120,7 +120,7 @@ static ssize_t syseeprom_read(struct fil
     return rd_len;
 }
 
-static ssize_t syseeprom_write(struct file *filp, struct kobject *kobj, struct bin_attribute *attr,
+static ssize_t syseeprom_write(struct file *filp, struct kobject *kobj, const struct bin_attribute *attr,
                    char *buf, loff_t offset, size_t count)
 {
     ssize_t wr_len;
Index: sonic-platform-modules-micas/common/modules/s3ip_sysfs/sysfs_driver/transceiver_sysfs.c
===================================================================
--- sonic-platform-modules-micas.orig/common/modules/s3ip_sysfs/sysfs_driver/transceiver_sysfs.c
+++ sonic-platform-modules-micas/common/modules/s3ip_sysfs/sysfs_driver/transceiver_sysfs.c
@@ -676,7 +676,7 @@ static ssize_t eth_interrupt_show(struct
     return g_sff_drv->get_eth_interrupt_status(eth_index, buf, PAGE_SIZE);
 }
 
-static ssize_t eth_eeprom_read(struct file *filp, struct kobject *kobj, struct bin_attribute *attr,
+static ssize_t eth_eeprom_read(struct file *filp, struct kobject *kobj, const struct bin_attribute *attr,
                                char *buf, loff_t offset, size_t count)
 {
     struct switch_obj *eth_obj;
@@ -702,7 +702,7 @@ static ssize_t eth_eeprom_read(struct fi
     return rd_len;
 }
 
-static ssize_t eth_eeprom_write(struct file *filp, struct kobject *kobj, struct bin_attribute *attr,
+static ssize_t eth_eeprom_write(struct file *filp, struct kobject *kobj, const struct bin_attribute *attr,
                                 char *buf, loff_t offset, size_t count)
 {
     struct switch_obj *eth_obj;
