Index: sonic-platform-modules-micas/common/modules/wb_csu550.c
===================================================================
--- sonic-platform-modules-micas.orig/common/modules/wb_csu550.c
+++ sonic-platform-modules-micas/common/modules/wb_csu550.c
@@ -257,4 +257,4 @@ module_i2c_driver(pmbus_driver);
 MODULE_AUTHOR("support");
 MODULE_DESCRIPTION("Generic PMBus driver");
 MODULE_LICENSE("GPL");
-MODULE_IMPORT_NS(PMBUS);
+MODULE_IMPORT_NS("PMBUS");
Index: sonic-platform-modules-micas/common/modules/wb_xdpe132g5c_pmbus.c
===================================================================
--- sonic-platform-modules-micas.orig/common/modules/wb_xdpe132g5c_pmbus.c
+++ sonic-platform-modules-micas/common/modules/wb_xdpe132g5c_pmbus.c
@@ -421,4 +421,4 @@ module_i2c_driver(xdpe132g5c_driver);
 MODULE_AUTHOR("support");
 MODULE_DESCRIPTION("PMBus driver for Infineon XDPE132g5 family");
 MODULE_LICENSE("GPL");
-MODULE_IMPORT_NS(PMBUS);
+MODULE_IMPORT_NS("PMBUS");
