Index: sonic-platform-modules-micas/debian/control
===================================================================
--- sonic-platform-modules-micas.orig/debian/control
+++ sonic-platform-modules-micas/debian/control
@@ -1,7 +1,7 @@
 Source: sonic-micas-platform-modules
 Section: main
 Priority: extra
-Maintainer: support
+Maintainer: support <support@micasnetworks.com>
 Standards-Version: 3.9.3
 
 Package: platform-modules-micas-m2-w6510-48v8c
