Index: sonic-platform-modules-micas/common/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/common/Makefile
+++ sonic-platform-modules-micas/common/Makefile
@@ -3,8 +3,8 @@ CC ?=gcc
 INSTALL_MOD_DIR ?=extra
 KVERSION   ?= $(shell uname -r)
 KERNEL_SRC ?=  /lib/modules/$(KVERSION)
-EXTRA_CFLAGS:= -I$(M)/include
-EXTRA_CFLAGS+= -Wall
+ccflags-y:= -I$(M)/include
+ccflags-y+= -Wall
 SUB_BUILD_DIR = $(PWD)/build
 DIR_KERNEL_SRC = $(PWD)/modules
 SCRIPT_DIR = $(PWD)/script
Index: sonic-platform-modules-micas/common/modules/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/common/modules/Makefile
+++ sonic-platform-modules-micas/common/modules/Makefile
@@ -1,7 +1,7 @@
 PWD		= $(shell pwd)
-EXTRA_CFLAGS:= -I$(M)/include
-EXTRA_CFLAGS+= -I$(M)
-EXTRA_CFLAGS+= -Wall
+ccflags-y:= -I$(M)/include
+ccflags-y+= -I$(M)
+ccflags-y+= -Wall
 KVERSION   ?= $(shell uname -r)
 KERNEL_SRC ?=  /lib/modules/$(KVERSION)
 
Index: sonic-platform-modules-micas/common/modules/plat_sysfs/dev_cfg/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/common/modules/plat_sysfs/dev_cfg/Makefile
+++ sonic-platform-modules-micas/common/modules/plat_sysfs/dev_cfg/Makefile
@@ -1,7 +1,7 @@
 PWD		= $(shell pwd)
 
-EXTRA_CFLAGS:= -I$(M)/include
-EXTRA_CFLAGS+= -Wall
+ccflags-y:= -I$(M)/include
+ccflags-y+= -Wall
 SUBDIR_CFG = cfg
 plat_dfd-objs := dfd_module.o dfd_fan_driver.o \
 dfd_slot_driver.o \
Index: sonic-platform-modules-micas/common/modules/plat_sysfs/dev_sysfs/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/common/modules/plat_sysfs/dev_sysfs/Makefile
+++ sonic-platform-modules-micas/common/modules/plat_sysfs/dev_sysfs/Makefile
@@ -1,7 +1,7 @@
 PWD		= $(shell pwd)
 
-EXTRA_CFLAGS:= -I$(M)/include
-EXTRA_CFLAGS+= -Wall
+ccflags-y:= -I$(M)/include
+ccflags-y+= -Wall
 KBUILD_EXTRA_SYMBOLS += $(PLAT_SYSFS_DIR)/dev_cfg/Module.symvers
 
 obj-m := plat_switch.o
Index: sonic-platform-modules-micas/common/modules/s3ip_sysfs/device_driver/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/common/modules/s3ip_sysfs/device_driver/Makefile
+++ sonic-platform-modules-micas/common/modules/s3ip_sysfs/device_driver/Makefile
@@ -3,10 +3,10 @@ PWD		= $(shell pwd)
 MAKEFILE_FILE_PATH = $(abspath $(lastword $(MAKEFILE_LIST)))
 DEV_SYSFS_HEADER_DIR = $(abspath $(MAKEFILE_FILE_PATH)/../../sysfs_driver/include)
 SWITCH_DVR_HEADER_DIR = $(abspath $(MAKEFILE_FILE_PATH)/../../switch_driver/include)
-EXTRA_CFLAGS:= -I$(M)/include
-EXTRA_CFLAGS+= -I$(DEV_SYSFS_HEADER_DIR)
-EXTRA_CFLAGS+= -I$(SWITCH_DVR_HEADER_DIR)
-EXTRA_CFLAGS+= -Wall
+ccflags-y:= -I$(M)/include
+ccflags-y+= -I$(DEV_SYSFS_HEADER_DIR)
+ccflags-y+= -I$(SWITCH_DVR_HEADER_DIR)
+ccflags-y+= -Wall
 
 KBUILD_EXTRA_SYMBOLS += $(SYSFS_DRIVER_DIR)/Module.symvers
 KBUILD_EXTRA_SYMBOLS += $(SWITCH_DRIVER_DIR)/Module.symvers
Index: sonic-platform-modules-micas/common/modules/s3ip_sysfs/switch_driver/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/common/modules/s3ip_sysfs/switch_driver/Makefile
+++ sonic-platform-modules-micas/common/modules/s3ip_sysfs/switch_driver/Makefile
@@ -1,7 +1,7 @@
 PWD		= $(shell pwd)
 
-EXTRA_CFLAGS:= -I$(M)/include
-EXTRA_CFLAGS+= -Wall
+ccflags-y:= -I$(M)/include
+ccflags-y+= -Wall
 
 SUBDIR_CFG = cfg
 wb_switch_driver-objs := switch_driver.o wb_module.o \
Index: sonic-platform-modules-micas/common/modules/s3ip_sysfs/sysfs_driver/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/common/modules/s3ip_sysfs/sysfs_driver/Makefile
+++ sonic-platform-modules-micas/common/modules/s3ip_sysfs/sysfs_driver/Makefile
@@ -3,10 +3,10 @@ PWD		= $(shell pwd)
 MAKEFILE_FILE_PATH = $(abspath $(lastword $(MAKEFILE_LIST)))
 DEV_SYSFS_HEADER_DIR = $(abspath $(MAKEFILE_FILE_PATH)/../../sysfs_driver/include)
 SWITCH_DVR_HEADER_DIR = $(abspath $(MAKEFILE_FILE_PATH)/../../switch_driver/include)
-EXTRA_CFLAGS:= -I$(M)/include
-EXTRA_CFLAGS+= -I$(DEV_SYSFS_HEADER_DIR)
-EXTRA_CFLAGS+= -I$(SWITCH_DVR_HEADER_DIR)
-EXTRA_CFLAGS+= -Wall
+ccflags-y:= -I$(M)/include
+ccflags-y+= -I$(DEV_SYSFS_HEADER_DIR)
+ccflags-y+= -I$(SWITCH_DVR_HEADER_DIR)
+ccflags-y+= -Wall
 
 s3ip_sysfs-objs := switch.o cpld_sysfs.o \
 curr_sensor_sysfs.o \
Index: sonic-platform-modules-micas/m2-w6510-32c/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/m2-w6510-32c/Makefile
+++ sonic-platform-modules-micas/m2-w6510-32c/Makefile
@@ -1,7 +1,7 @@
 PWD		= $(shell pwd)
 DIR_KERNEL_SRC = $(PWD)/modules/driver
-EXTRA_CFLAGS:= -I$(M)/include
-EXTRA_CFLAGS+= -Wall
+ccflags-y:= -I$(M)/include
+ccflags-y+= -Wall
 SUB_BUILD_DIR = $(PWD)/build
 INSTALL_DIR = $(SUB_BUILD_DIR)/$(KERNEL_SRC)/$(INSTALL_MOD_DIR)
 INSTALL_SCRIPT_DIR = $(SUB_BUILD_DIR)/usr/local/bin
Index: sonic-platform-modules-micas/m2-w6510-32c/modules/driver/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/m2-w6510-32c/modules/driver/Makefile
+++ sonic-platform-modules-micas/m2-w6510-32c/modules/driver/Makefile
@@ -1,7 +1,7 @@
 MAKEFILE_FILE_PATH = $(abspath $(lastword $(MAKEFILE_LIST)))
 MODULES_DIR = $(abspath $(MAKEFILE_FILE_PATH)/../../../../common/modules)
 
-EXTRA_CFLAGS+= -I$(MODULES_DIR)
+ccflags-y+= -I$(MODULES_DIR)
 
 obj-m := wb_pcie_dev_device.o
 obj-m += wb_i2c_mux_pca954x_device.o
Index: sonic-platform-modules-micas/m2-w6510-48gt4v/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/m2-w6510-48gt4v/Makefile
+++ sonic-platform-modules-micas/m2-w6510-48gt4v/Makefile
@@ -1,7 +1,7 @@
 PWD		= $(shell pwd)
 DIR_KERNEL_SRC = $(PWD)/modules/driver
-EXTRA_CFLAGS:= -I$(M)/include
-EXTRA_CFLAGS+= -Wall
+ccflags-y:= -I$(M)/include
+ccflags-y+= -Wall
 SUB_BUILD_DIR = $(PWD)/build
 INSTALL_DIR = $(SUB_BUILD_DIR)/$(KERNEL_SRC)/$(INSTALL_MOD_DIR)
 INSTALL_SCRIPT_DIR = $(SUB_BUILD_DIR)/usr/local/bin
Index: sonic-platform-modules-micas/m2-w6510-48gt4v/modules/driver/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/m2-w6510-48gt4v/modules/driver/Makefile
+++ sonic-platform-modules-micas/m2-w6510-48gt4v/modules/driver/Makefile
@@ -1,7 +1,7 @@
 MAKEFILE_FILE_PATH = $(abspath $(lastword $(MAKEFILE_LIST)))
 MODULES_DIR = $(abspath $(MAKEFILE_FILE_PATH)/../../../../common/modules)
 
-EXTRA_CFLAGS+= -I$(MODULES_DIR)
+ccflags-y+= -I$(MODULES_DIR)
 
 obj-m += wb_i2c_mux_pca954x_device.o
 obj-m += wb_lpc_drv_device.o
Index: sonic-platform-modules-micas/m2-w6510-48v8c/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/m2-w6510-48v8c/Makefile
+++ sonic-platform-modules-micas/m2-w6510-48v8c/Makefile
@@ -1,7 +1,7 @@
 PWD		= $(shell pwd)
 DIR_KERNEL_SRC = $(PWD)/modules/driver
-EXTRA_CFLAGS:= -I$(M)/include
-EXTRA_CFLAGS+= -Wall
+ccflags-y:= -I$(M)/include
+ccflags-y+= -Wall
 SUB_BUILD_DIR = $(PWD)/build
 INSTALL_DIR = $(SUB_BUILD_DIR)/$(KERNEL_SRC)/$(INSTALL_MOD_DIR)
 INSTALL_SCRIPT_DIR = $(SUB_BUILD_DIR)/usr/local/bin
Index: sonic-platform-modules-micas/m2-w6510-48v8c/modules/driver/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/m2-w6510-48v8c/modules/driver/Makefile
+++ sonic-platform-modules-micas/m2-w6510-48v8c/modules/driver/Makefile
@@ -1,7 +1,7 @@
 MAKEFILE_FILE_PATH = $(abspath $(lastword $(MAKEFILE_LIST)))
 MODULES_DIR = $(abspath $(MAKEFILE_FILE_PATH)/../../../../common/modules)
 
-EXTRA_CFLAGS+= -I$(MODULES_DIR)
+ccflags-y+= -I$(MODULES_DIR)
 
 obj-m :=  wb_pcie_dev_device.o
 obj-m += wb_i2c_mux_pca954x_device.o
Index: sonic-platform-modules-micas/m2-w6520-24dc8qc/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/m2-w6520-24dc8qc/Makefile
+++ sonic-platform-modules-micas/m2-w6520-24dc8qc/Makefile
@@ -1,7 +1,7 @@
 PWD		= $(shell pwd)
 DIR_KERNEL_SRC = $(PWD)/modules/driver
-EXTRA_CFLAGS:= -I$(M)/include
-EXTRA_CFLAGS+= -Wall
+ccflags-y:= -I$(M)/include
+ccflags-y+= -Wall
 SUB_BUILD_DIR = $(PWD)/build
 INSTALL_DIR = $(SUB_BUILD_DIR)/$(KERNEL_SRC)/$(INSTALL_MOD_DIR)
 INSTALL_SCRIPT_DIR = $(SUB_BUILD_DIR)/usr/local/bin
Index: sonic-platform-modules-micas/m2-w6520-24dc8qc/modules/driver/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/m2-w6520-24dc8qc/modules/driver/Makefile
+++ sonic-platform-modules-micas/m2-w6520-24dc8qc/modules/driver/Makefile
@@ -1,7 +1,7 @@
 MAKEFILE_FILE_PATH = $(abspath $(lastword $(MAKEFILE_LIST)))
 MODULES_DIR = $(abspath $(MAKEFILE_FILE_PATH)/../../../../common/modules)
 
-EXTRA_CFLAGS+= -I$(MODULES_DIR)
+ccflags-y+= -I$(MODULES_DIR)
 
 obj-m := wb_pcie_dev_device.o
 obj-m += wb_fpga_pca954x_device.o
Index: sonic-platform-modules-micas/m2-w6520-48c8qc/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/m2-w6520-48c8qc/Makefile
+++ sonic-platform-modules-micas/m2-w6520-48c8qc/Makefile
@@ -1,7 +1,7 @@
 PWD		= $(shell pwd)
 DIR_KERNEL_SRC = $(PWD)/modules/driver
-EXTRA_CFLAGS:= -I$(M)/include
-EXTRA_CFLAGS+= -Wall
+ccflags-y:= -I$(M)/include
+ccflags-y+= -Wall
 SUB_BUILD_DIR = $(PWD)/build
 INSTALL_DIR = $(SUB_BUILD_DIR)/$(KERNEL_SRC)/$(INSTALL_MOD_DIR)
 INSTALL_SCRIPT_DIR = $(SUB_BUILD_DIR)/usr/local/bin
Index: sonic-platform-modules-micas/m2-w6520-48c8qc/modules/driver/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/m2-w6520-48c8qc/modules/driver/Makefile
+++ sonic-platform-modules-micas/m2-w6520-48c8qc/modules/driver/Makefile
@@ -1,6 +1,6 @@
 MAKEFILE_FILE_PATH = $(abspath $(lastword $(MAKEFILE_LIST)))
 MODULES_DIR = $(abspath $(MAKEFILE_FILE_PATH)/../../../../common/modules)
-EXTRA_CFLAGS+= -I$(MODULES_DIR)
+ccflags-y+= -I$(MODULES_DIR)
 
 obj-m := wb_pcie_dev_device.o
 obj-m += wb_fpga_pca954x_device.o
Index: sonic-platform-modules-micas/m2-w6920-32qc2x/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/m2-w6920-32qc2x/Makefile
+++ sonic-platform-modules-micas/m2-w6920-32qc2x/Makefile
@@ -1,7 +1,7 @@
 PWD		= $(shell pwd)
 DIR_KERNEL_SRC = $(PWD)/modules/driver
-EXTRA_CFLAGS:= -I$(M)/include
-EXTRA_CFLAGS+= -Wall
+ccflags-y:= -I$(M)/include
+ccflags-y+= -Wall
 SUB_BUILD_DIR = $(PWD)/build
 INSTALL_DIR = $(SUB_BUILD_DIR)/$(KERNEL_SRC)/$(INSTALL_MOD_DIR)
 INSTALL_SCRIPT_DIR = $(SUB_BUILD_DIR)/usr/local/bin
Index: sonic-platform-modules-micas/m2-w6920-32qc2x/modules/driver/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/m2-w6920-32qc2x/modules/driver/Makefile
+++ sonic-platform-modules-micas/m2-w6920-32qc2x/modules/driver/Makefile
@@ -1,7 +1,7 @@
 MAKEFILE_FILE_PATH = $(abspath $(lastword $(MAKEFILE_LIST)))
 MODULES_DIR = $(abspath $(MAKEFILE_FILE_PATH)/../../../../common/modules)
 
-EXTRA_CFLAGS+= -I$(MODULES_DIR)
+ccflags-y+= -I$(MODULES_DIR)
 
 
 obj-m += wb_pcie_dev_device.o
Index: sonic-platform-modules-micas/m2-w6930-64qc/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/m2-w6930-64qc/Makefile
+++ sonic-platform-modules-micas/m2-w6930-64qc/Makefile
@@ -1,7 +1,7 @@
 PWD		= $(shell pwd)
 DIR_KERNEL_SRC = $(PWD)/modules/driver
-EXTRA_CFLAGS:= -I$(M)/include
-EXTRA_CFLAGS+= -Wall
+ccflags-y:= -I$(M)/include
+ccflags-y+= -Wall
 SUB_BUILD_DIR = $(PWD)/build
 INSTALL_DIR = $(SUB_BUILD_DIR)/$(KERNEL_SRC)/$(INSTALL_MOD_DIR)
 INSTALL_SCRIPT_DIR = $(SUB_BUILD_DIR)/usr/local/bin
Index: sonic-platform-modules-micas/m2-w6930-64qc/modules/driver/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/m2-w6930-64qc/modules/driver/Makefile
+++ sonic-platform-modules-micas/m2-w6930-64qc/modules/driver/Makefile
@@ -1,7 +1,7 @@
 MAKEFILE_FILE_PATH = $(abspath $(lastword $(MAKEFILE_LIST)))
 MODULES_DIR = $(abspath $(MAKEFILE_FILE_PATH)/../../../../common/modules)
 
-EXTRA_CFLAGS+= -I$(MODULES_DIR)
+ccflags-y+= -I$(MODULES_DIR)
 
 obj-m += wb_pcie_dev_device.o
 obj-m += wb_spi_ocores_device.o
Index: sonic-platform-modules-micas/m2-w6940-128qc/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/m2-w6940-128qc/Makefile
+++ sonic-platform-modules-micas/m2-w6940-128qc/Makefile
@@ -1,7 +1,7 @@
 PWD		= $(shell pwd)
 DIR_KERNEL_SRC = $(PWD)/modules/driver
-EXTRA_CFLAGS:= -I$(M)/include
-EXTRA_CFLAGS+= -Wall
+ccflags-y:= -I$(M)/include
+ccflags-y+= -Wall
 SUB_BUILD_DIR = $(PWD)/build
 INSTALL_DIR = $(SUB_BUILD_DIR)/$(KERNEL_SRC)/$(INSTALL_MOD_DIR)
 INSTALL_SCRIPT_DIR = $(SUB_BUILD_DIR)/usr/local/bin
Index: sonic-platform-modules-micas/m2-w6940-128qc/modules/driver/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/m2-w6940-128qc/modules/driver/Makefile
+++ sonic-platform-modules-micas/m2-w6940-128qc/modules/driver/Makefile
@@ -1,7 +1,7 @@
 MAKEFILE_FILE_PATH = $(abspath $(lastword $(MAKEFILE_LIST)))
 MODULES_DIR = $(abspath $(MAKEFILE_FILE_PATH)/../../../../common/modules)
 
-EXTRA_CFLAGS+= -I$(MODULES_DIR)
+ccflags-y+= -I$(MODULES_DIR)
 
 obj-m := wb_pcie_dev_device.o
 obj-m += wb_i2c_ocores_device.o
Index: sonic-platform-modules-micas/m2-w6940-128x1-fr4-pure-pd/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/m2-w6940-128x1-fr4-pure-pd/Makefile
+++ sonic-platform-modules-micas/m2-w6940-128x1-fr4-pure-pd/Makefile
@@ -1,7 +1,7 @@
 PWD		= $(shell pwd)
 DIR_KERNEL_SRC = $(PWD)/modules/driver
-EXTRA_CFLAGS:= -I$(M)/include
-EXTRA_CFLAGS+= -Wall
+ccflags-y:= -I$(M)/include
+ccflags-y+= -Wall
 SUB_BUILD_DIR = $(PWD)/build
 INSTALL_DIR = $(SUB_BUILD_DIR)/$(KERNEL_SRC)/$(INSTALL_MOD_DIR)
 INSTALL_SCRIPT_DIR = $(SUB_BUILD_DIR)/usr/local/bin
Index: sonic-platform-modules-micas/m2-w6940-128x1-fr4-pure-pd/modules/driver/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/m2-w6940-128x1-fr4-pure-pd/modules/driver/Makefile
+++ sonic-platform-modules-micas/m2-w6940-128x1-fr4-pure-pd/modules/driver/Makefile
@@ -3,8 +3,8 @@ MAKEFILE_DIR = $(dir $(MAKEFILE_FILE_PAT
 MODULES_DIR = $(abspath $(MAKEFILE_DIR)/../../../common/modules)
 BSP_PUBLIC_PATH = $(abspath $(MAKEFILE_DIR)/../../../public/include)
 
-EXTRA_CFLAGS+= -I$(BSP_PUBLIC_PATH)
-EXTRA_CFLAGS+= -I$(MODULES_DIR)
+ccflags-y+= -I$(BSP_PUBLIC_PATH)
+ccflags-y+= -I$(MODULES_DIR)
 
 obj-m := wb_pcie_dev_device.o
 obj-m += wb_i2c_ocores_device.o
Index: sonic-platform-modules-micas/m2-w6940-128x1-fr4/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/m2-w6940-128x1-fr4/Makefile
+++ sonic-platform-modules-micas/m2-w6940-128x1-fr4/Makefile
@@ -1,7 +1,7 @@
 PWD		= $(shell pwd)
 DIR_KERNEL_SRC = $(PWD)/modules/driver
-EXTRA_CFLAGS:= -I$(M)/include
-EXTRA_CFLAGS+= -Wall
+ccflags-y:= -I$(M)/include
+ccflags-y+= -Wall
 SUB_BUILD_DIR = $(PWD)/build
 INSTALL_DIR = $(SUB_BUILD_DIR)/$(KERNEL_SRC)/$(INSTALL_MOD_DIR)
 INSTALL_SCRIPT_DIR = $(SUB_BUILD_DIR)/usr/local/bin
Index: sonic-platform-modules-micas/m2-w6940-128x1-fr4/modules/driver/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/m2-w6940-128x1-fr4/modules/driver/Makefile
+++ sonic-platform-modules-micas/m2-w6940-128x1-fr4/modules/driver/Makefile
@@ -3,8 +3,8 @@ MAKEFILE_DIR = $(dir $(MAKEFILE_FILE_PAT
 MODULES_DIR = $(abspath $(MAKEFILE_DIR)/../../../common/modules)
 BSP_PUBLIC_PATH = $(abspath $(MAKEFILE_DIR)/../../../public/include)
 
-EXTRA_CFLAGS+= -I$(BSP_PUBLIC_PATH)
-EXTRA_CFLAGS+= -I$(MODULES_DIR)
+ccflags-y+= -I$(BSP_PUBLIC_PATH)
+ccflags-y+= -I$(MODULES_DIR)
 
 obj-m := wb_pcie_dev_device.o
 obj-m += wb_i2c_ocores_device.o
Index: sonic-platform-modules-micas/m2-w6940-64oc/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/m2-w6940-64oc/Makefile
+++ sonic-platform-modules-micas/m2-w6940-64oc/Makefile
@@ -1,7 +1,7 @@
 PWD		= $(shell pwd)
 DIR_KERNEL_SRC = $(PWD)/modules/driver
-EXTRA_CFLAGS:= -I$(M)/include
-EXTRA_CFLAGS+= -Wall
+ccflags-y:= -I$(M)/include
+ccflags-y+= -Wall
 SUB_BUILD_DIR = $(PWD)/build
 INSTALL_DIR = $(SUB_BUILD_DIR)/$(KERNEL_SRC)/$(INSTALL_MOD_DIR)
 INSTALL_SCRIPT_DIR = $(SUB_BUILD_DIR)/usr/local/bin
Index: sonic-platform-modules-micas/m2-w6940-64oc/modules/driver/Makefile
===================================================================
--- sonic-platform-modules-micas.orig/m2-w6940-64oc/modules/driver/Makefile
+++ sonic-platform-modules-micas/m2-w6940-64oc/modules/driver/Makefile
@@ -1,7 +1,7 @@
 MAKEFILE_FILE_PATH = $(abspath $(lastword $(MAKEFILE_LIST)))
 MODULES_DIR = $(abspath $(MAKEFILE_FILE_PATH)/../../../../common/modules)
 
-EXTRA_CFLAGS+= -I$(MODULES_DIR)
+ccflags-y+= -I$(MODULES_DIR)
 
 obj-m += wb_pcie_dev_device.o
 obj-m += wb_io_dev_device.o
