Index: saibcm-modules-legacy-th/debian/opennsl-modules.dirs
===================================================================
--- saibcm-modules-legacy-th.orig/debian/opennsl-modules.dirs
+++ saibcm-modules-legacy-th/debian/opennsl-modules.dirs
@@ -1 +1 @@
-lib/modules/6.12.41+deb13-sonic-amd64/extra
+lib/modules/7.0.0-1002-sonic/extra
Index: saibcm-modules-legacy-th/debian/opennsl-modules.install
===================================================================
--- saibcm-modules-legacy-th.orig/debian/opennsl-modules.install
+++ saibcm-modules-legacy-th/debian/opennsl-modules.install
@@ -1,8 +1,8 @@
-systems/linux/user/x86-smp_generic_64-2_6/linux-bcm-knet.ko lib/modules/6.12.41+deb13-sonic-amd64/extra
-systems/linux/user/x86-smp_generic_64-2_6/linux-kernel-bde.ko lib/modules/6.12.41+deb13-sonic-amd64/extra
-systems/linux/user/x86-smp_generic_64-2_6/linux-user-bde.ko lib/modules/6.12.41+deb13-sonic-amd64/extra
-systems/linux/user/x86-smp_generic_64-2_6/linux-knet-cb.ko lib/modules/6.12.41+deb13-sonic-amd64/extra
+systems/linux/user/x86-smp_generic_64-2_6/linux-bcm-knet.ko lib/modules/7.0.0-1002-sonic/extra
+systems/linux/user/x86-smp_generic_64-2_6/linux-kernel-bde.ko lib/modules/7.0.0-1002-sonic/extra
+systems/linux/user/x86-smp_generic_64-2_6/linux-user-bde.ko lib/modules/7.0.0-1002-sonic/extra
+systems/linux/user/x86-smp_generic_64-2_6/linux-knet-cb.ko lib/modules/7.0.0-1002-sonic/extra
 systemd/opennsl-modules.service lib/systemd/system
-sdklt/build/bde/linux_ngbde.ko lib/modules/6.12.41+deb13-sonic-amd64/extra
-sdklt/build/knet/linux_ngknet.ko lib/modules/6.12.41+deb13-sonic-amd64/extra
-sdklt/build/knetcb/linux_ngknetcb.ko lib/modules/6.12.41+deb13-sonic-amd64/extra
+sdklt/build/bde/linux_ngbde.ko lib/modules/7.0.0-1002-sonic/extra
+sdklt/build/knet/linux_ngknet.ko lib/modules/7.0.0-1002-sonic/extra
+sdklt/build/knetcb/linux_ngknetcb.ko lib/modules/7.0.0-1002-sonic/extra
