Index: saibcm-modules-legacy-th/sdklt/Makefile
===================================================================
--- saibcm-modules-legacy-th.orig/sdklt/Makefile
+++ saibcm-modules-legacy-th/sdklt/Makefile
@@ -70,36 +70,36 @@ kmod: $(KMODS)
 bde:
 	$(MAKE) -C $(SDK)/linux/bde SDK=$(SDK) \
 		LKM_BLDDIR=$(BDE_BLDDIR) $(TARGET)
-	ln -sf $(BDE_BLDDIR)/*.ko $(OUTPUT_DIR)
+	ln -sf $(BDE_BLDDIR)/*.ko $(OUTPUT_DIR) 2>/dev/null || true
 
 knet: bde
 	$(MAKE) -C $(SDK)/linux/knet SDK=$(SDK) \
 		KBUILD_EXTRA_SYMBOLS=$(BDE_BLDDIR)/Module.symvers \
 		LKM_BLDDIR=$(KNET_BLDDIR) $(TARGET)
-	ln -sf $(KNET_BLDDIR)/*.ko $(OUTPUT_DIR)
+	ln -sf $(KNET_BLDDIR)/*.ko $(OUTPUT_DIR) 2>/dev/null || true
 
 knetcb: knet
 	$(MAKE) -C $(SDK)/linux/knetcb SDK=$(SDK) \
 		KBUILD_EXTRA_SYMBOLS=$(KNET_BLDDIR)/Module.symvers \
 		LKM_BLDDIR=$(KNETCB_BLDDIR) $(TARGET)
-	ln -sf $(KNETCB_BLDDIR)/*.ko $(OUTPUT_DIR)
+	ln -sf $(KNETCB_BLDDIR)/*.ko $(OUTPUT_DIR) 2>/dev/null || true
 
 bcmgenl: knet genl-packet
 	$(MAKE) -C $(SDK)/linux/bcmgenl SDK=$(SDK) \
 		KBUILD_EXTRA_SYMBOLS=$(KNET_BLDDIR)/Module.symvers \
 		LKM_BLDDIR=$(BCMGENL_BLDDIR) $(TARGET)
-	ln -sf $(BCMGENL_BLDDIR)/*.ko $(OUTPUT_DIR)
+	ln -sf $(BCMGENL_BLDDIR)/*.ko $(OUTPUT_DIR) 2>/dev/null || true
 
 genl-packet:
 	$(MAKE) -C $(SDK)/linux/genl-packet SDK=$(SDK) \
 		LKM_BLDDIR=$(GENL_PACKET_BLDDIR) $(TARGET)
-	ln -sf $(GENL_PACKET_BLDDIR)/*.ko $(OUTPUT_DIR)
+	ln -sf $(GENL_PACKET_BLDDIR)/*.ko $(OUTPUT_DIR) 2>/dev/null || true
 
 edk: bde
 	$(MAKE) -C $(SDK)/linux/edk SDK=$(SDK) \
 		KBUILD_EXTRA_SYMBOLS=$(BDE_BLDDIR)/Module.symvers \
 		LKM_BLDDIR=$(EDK_BLDDIR) $(TARGET)
-	ln -sf $(EDK_BLDDIR)/*.ko $(OUTPUT_DIR)
+	ln -sf $(EDK_BLDDIR)/*.ko $(OUTPUT_DIR) 2>/dev/null || true
 
 ptpclock: bde knet edk
 	$(MAKE) -C $(SDK)/linux/ptpclock SDK=$(SDK) \
@@ -107,13 +107,13 @@ ptpclock: bde knet edk
 		KBUILD_EXTRA_SYMBOLS+=$(KNET_BLDDIR)/Module.symvers \
 		KBUILD_EXTRA_SYMBOLS+=$(EDK_BLDDIR)/Module.symvers \
 		LKM_BLDDIR=$(PTPCLOCK_BLDDIR) $(TARGET)
-	ln -sf $(PTPCLOCK_BLDDIR)/*.ko $(OUTPUT_DIR)
+	ln -sf $(PTPCLOCK_BLDDIR)/*.ko $(OUTPUT_DIR) 2>/dev/null || true
 
 st: bde
 	$(MAKE) -C $(SDK)/linux/st SDK=$(SDK) \
 		KBUILD_EXTRA_SYMBOLS=$(BDE_BLDDIR)/Module.symvers \
 		LKM_BLDDIR=$(ST_BLDDIR) $(TARGET)
-	ln -sf $(ST_BLDDIR)/*.ko $(OUTPUT_DIR)
+	ln -sf $(ST_BLDDIR)/*.ko $(OUTPUT_DIR) 2>/dev/null || true
 
 clean:
 	$(MAKE) kmod TARGET=$@
