Index: sonic-platform-modules-nokia/mackinac/modules/cpuctl_spi.c
===================================================================
--- sonic-platform-modules-nokia.orig/mackinac/modules/cpuctl_spi.c
+++ sonic-platform-modules-nokia/mackinac/modules/cpuctl_spi.c
@@ -339,7 +339,7 @@
     int rc = 0, i;
     int bus_num = pdev->ctlv->spi_bus;
 
-    spicon = devm_spi_alloc_master(dev, sizeof(CTLDEV));
+    spicon = devm_spi_alloc_target(dev, sizeof(CTLDEV));
     if (!spicon)
         return -ENOMEM;
 
