Index: sonic-platform-modules-nokia/ixr7220h4-32d/modules/fpga_gpio.c
===================================================================
--- sonic-platform-modules-nokia.orig/ixr7220h4-32d/modules/fpga_gpio.c
+++ sonic-platform-modules-nokia/ixr7220h4-32d/modules/fpga_gpio.c
@@ -46,7 +46,7 @@ static int fpga_gpio_get(struct gpio_chi
     return ret;
 }
 
-static void fpga_gpio_set(struct gpio_chip *gc,
+static int fpga_gpio_set(struct gpio_chip *gc,
                           unsigned gpio, int val)
 {
     struct fpga_gpio_chip *chip = gpiochip_get_data(gc);
@@ -66,6 +66,8 @@ static void fpga_gpio_set(struct gpio_ch
     iowrite32(wdata, chip->bar + reg);
 
     mutex_unlock(&chip->lock);
+
+    return 0;
 }
 
 int gpiodev_init(struct pci_dev *dev, struct fpga_dev *fpga)
Index: sonic-platform-modules-nokia/ixr7220h5-32d/modules/fpga_gpio.c
===================================================================
--- sonic-platform-modules-nokia.orig/ixr7220h5-32d/modules/fpga_gpio.c
+++ sonic-platform-modules-nokia/ixr7220h5-32d/modules/fpga_gpio.c
@@ -65,7 +65,7 @@ static int fpga_gpio_get(struct gpio_chi
     return ret;
 }
 
-static void fpga_gpio_set(struct gpio_chip *gc,
+static int fpga_gpio_set(struct gpio_chip *gc,
                           unsigned gpio, int val)
 {
     struct fpga_gpio_chip *chip = gpiochip_get_data(gc);
@@ -85,6 +85,8 @@ static void fpga_gpio_set(struct gpio_ch
     iowrite32(wdata, chip->bar + reg);
 
     mutex_unlock(&chip->lock);
+
+    return 0;
 }
 
 int gpiodev_init(struct pci_dev *dev, struct fpga_dev *fpga)
Index: sonic-platform-modules-nokia/ixr7220h5-64d/modules/fpga_gpio.c
===================================================================
--- sonic-platform-modules-nokia.orig/ixr7220h5-64d/modules/fpga_gpio.c
+++ sonic-platform-modules-nokia/ixr7220h5-64d/modules/fpga_gpio.c
@@ -65,7 +65,7 @@ static int fpga_gpio_get(struct gpio_chi
     return ret;
 }
 
-static void fpga_gpio_set(struct gpio_chip *gc,
+static int fpga_gpio_set(struct gpio_chip *gc,
                           unsigned gpio, int val)
 {
     struct fpga_gpio_chip *chip = gpiochip_get_data(gc);
@@ -85,6 +85,8 @@ static void fpga_gpio_set(struct gpio_ch
     iowrite32(wdata, chip->bar + reg);
 
     mutex_unlock(&chip->lock);
+
+    return 0;
 }
 
 int gpiodev_init(struct pci_dev *dev, struct fpga_dev *fpga)
Index: sonic-platform-modules-nokia/ixr7220h5-64o/modules/fpga_gpio.c
===================================================================
--- sonic-platform-modules-nokia.orig/ixr7220h5-64o/modules/fpga_gpio.c
+++ sonic-platform-modules-nokia/ixr7220h5-64o/modules/fpga_gpio.c
@@ -65,7 +65,7 @@ static int fpga_gpio_get(struct gpio_chi
     return ret;
 }
 
-static void fpga_gpio_set(struct gpio_chip *gc,
+static int fpga_gpio_set(struct gpio_chip *gc,
                           unsigned gpio, int val)
 {
     struct fpga_gpio_chip *chip = gpiochip_get_data(gc);
@@ -85,6 +85,8 @@ static void fpga_gpio_set(struct gpio_ch
     iowrite32(wdata, chip->bar + reg);
 
     mutex_unlock(&chip->lock);
+
+    return 0;
 }
 
 int gpiodev_init(struct pci_dev *dev, struct fpga_dev *fpga)
