Index: sonic-platform-modules-ruijie/common/modules/i2c-mux-pca954x.c
===================================================================
--- sonic-platform-modules-ruijie.orig/common/modules/i2c-mux-pca954x.c
+++ sonic-platform-modules-ruijie/common/modules/i2c-mux-pca954x.c
@@ -1139,15 +1139,15 @@ static int pca954x_do_gpio_reset(pca9548
         udelay(cfg_info->rst_delay_b);
         /* reset on */
         PCA954X_DEBUG("set gpio%d %d.\n", tmp_gpio_attr->gpio, tmp_gpio_attr->reset_on);
-        __gpio_set_value(tmp_gpio_attr->gpio, tmp_gpio_attr->reset_on);
+        gpio_set_value(tmp_gpio_attr->gpio, tmp_gpio_attr->reset_on);
         udelay(cfg_info->rst_delay);
         /* reset off */
         PCA954X_DEBUG("set gpio%d %d.\n", tmp_gpio_attr->gpio, tmp_gpio_attr->reset_off);
-        __gpio_set_value(tmp_gpio_attr->gpio, tmp_gpio_attr->reset_off);
+        gpio_set_value(tmp_gpio_attr->gpio, tmp_gpio_attr->reset_off);
 
         while (timeout > 0) {
             udelay(1);
-            val = __gpio_get_value(tmp_gpio_attr->gpio);
+            val = gpio_get_value(tmp_gpio_attr->gpio);
             if (val == tmp_gpio_attr->reset_off) {
                 adapter = adap;
                 /* get bus info */
@@ -1521,7 +1521,7 @@ static irqreturn_t pca954x_irq_handler(i
 
 	for (i = 0; i < data->chip->nchans; i++) {
 		if (ret & BIT(PCA954X_IRQ_OFFSET + i)) {
-			child_irq = irq_linear_revmap(data->irq, i);
+			child_irq = irq_find_mapping(data->irq, i);
 			handle_nested_irq(child_irq);
 			handled++;
 		}
