Index: sonic-platform-modules-ruijie/b6510-48vs8cq/modules/driver/rg_cpld.c
===================================================================
--- sonic-platform-modules-ruijie.orig/b6510-48vs8cq/modules/driver/rg_cpld.c
+++ sonic-platform-modules-ruijie/b6510-48vs8cq/modules/driver/rg_cpld.c
@@ -496,7 +496,7 @@ static  const struct attribute_group *cp
 	return NULL;
 }
 
-static int cpld_probe(struct i2c_client *client, const struct i2c_device_id *id)
+static int cpld_probe(struct i2c_client *client)
 {
     struct cpld_data *data;
     int status;
@@ -549,7 +549,7 @@ error:
 
 }
 
-static int cpld_remove(struct i2c_client *client)
+static void cpld_remove(struct i2c_client *client)
 {
     struct cpld_data *data = i2c_get_clientdata(client);
     const struct attribute_group *sysfs_group, *hwmon_group;
@@ -574,7 +574,6 @@ static int cpld_remove(struct i2c_client
         DBG_DEBUG("=========(addr:0x%x, nr:%d) no hwmon_device_unregister \n", client->addr, client->adapter->nr);
     }   
     
-    return 0;
 }
 
 
Index: sonic-platform-modules-ruijie/common/modules/csu550.c
===================================================================
--- sonic-platform-modules-ruijie.orig/common/modules/csu550.c
+++ sonic-platform-modules-ruijie/common/modules/csu550.c
@@ -156,8 +156,9 @@ abort:
 	return ret;
 }
 
-static int pmbus_probe(struct i2c_client *client, const struct i2c_device_id *id)
+static int pmbus_probe(struct i2c_client *client)
 {
+	const struct i2c_device_id *id = i2c_match_id(pmbus_id, client);
 	struct pmbus_driver_info *info;
 	struct pmbus_platform_data *pdata = NULL;
 	struct device *dev = &client->dev;
Index: sonic-platform-modules-ruijie/common/modules/i2c-mux-pca954x.c
===================================================================
--- sonic-platform-modules-ruijie.orig/common/modules/i2c-mux-pca954x.c
+++ sonic-platform-modules-ruijie/common/modules/i2c-mux-pca954x.c
@@ -1620,9 +1620,9 @@ err_req_irq:
 /*
  * I2C init/probing/exit functions
  */
-static int pca954x_probe(struct i2c_client *client,
-			 const struct i2c_device_id *id)
+static int pca954x_probe(struct i2c_client *client)
 {
+	const struct i2c_device_id *id = i2c_match_id(pca954x_id, client);
 	struct i2c_adapter *adap = to_i2c_adapter(client->dev.parent);
 	struct pca954x_platform_data *pdata = dev_get_platdata(&client->dev);
 	struct device_node *of_node = client->dev.of_node;
@@ -1713,7 +1713,7 @@ fail_del_adapters:
 	return ret;
 }
 
-static int pca954x_remove(struct i2c_client *client)
+static void pca954x_remove(struct i2c_client *client)
 {
 	struct i2c_mux_core *muxc = i2c_get_clientdata(client);
 	struct pca954x *data = i2c_mux_priv(muxc);
@@ -1728,7 +1728,6 @@ static int pca954x_remove(struct i2c_cli
 	}
 
 	i2c_mux_del_adapters(muxc);
-	return 0;
 }
 
 #ifdef CONFIG_PM_SLEEP
Index: sonic-platform-modules-ruijie/common/modules/i2c-mux-pca9641.c
===================================================================
--- sonic-platform-modules-ruijie.orig/common/modules/i2c-mux-pca9641.c
+++ sonic-platform-modules-ruijie/common/modules/i2c-mux-pca9641.c
@@ -544,8 +544,7 @@ void i2c_unlock_adapter(struct i2c_adapt
 /*
  * I2C init/probing/exit functions
  */
-static int pca9541_probe(struct i2c_client *client,
-			 const struct i2c_device_id *id)
+static int pca9541_probe(struct i2c_client *client)
 {
 	struct i2c_adapter *adap = client->adapter;
 	struct pca954x_platform_data *pdata = dev_get_platdata(&client->dev);
@@ -620,12 +619,11 @@ static int pca9541_probe(struct i2c_clie
 
 }
 
-static int pca9541_remove(struct i2c_client *client)
+static void pca9541_remove(struct i2c_client *client)
 {
 	struct i2c_mux_core *muxc = i2c_get_clientdata(client);
 
 	i2c_mux_del_adapters(muxc);
-	return 0;
 }
 
 static struct i2c_driver pca9641_driver = {
Index: sonic-platform-modules-ruijie/common/modules/pmbus.h
===================================================================
--- sonic-platform-modules-ruijie.orig/common/modules/pmbus.h
+++ sonic-platform-modules-ruijie/common/modules/pmbus.h
@@ -418,7 +418,7 @@ bool pmbus_check_byte_register(struct i2
 bool pmbus_check_word_register(struct i2c_client *client, int page, int reg);
 int pmbus_do_probe(struct i2c_client *client, const struct i2c_device_id *id,
 		   struct pmbus_driver_info *info);
-int pmbus_do_remove(struct i2c_client *client);
+void pmbus_do_remove(struct i2c_client *client);
 const struct pmbus_driver_info *pmbus_get_driver_info(struct i2c_client
 						      *client);
 #endif /* PMBUS_H */
Index: sonic-platform-modules-ruijie/common/modules/rg_fan.c
===================================================================
--- sonic-platform-modules-ruijie.orig/common/modules/rg_fan.c
+++ sonic-platform-modules-ruijie/common/modules/rg_fan.c
@@ -218,7 +218,7 @@ static const struct attribute_group fan_
 	.attrs = fan_sysfs_attrs,
 };
 
-static int fan_probe(struct i2c_client *client, const struct i2c_device_id *id)
+static int fan_probe(struct i2c_client *client)
 {
 	struct fan_data *data;
 	int status;
@@ -242,10 +242,9 @@ static int fan_probe(struct i2c_client *
 	return 0;
 }
 
-static int fan_remove(struct i2c_client *client)
+static void fan_remove(struct i2c_client *client)
 {
 	sysfs_remove_group(&client->dev.kobj, &fan_sysfs_group);
-	return 0;
 }
 
 static const struct i2c_device_id fan_id[] = { { "rg_fan", 0 }, {} };
Index: sonic-platform-modules-ruijie/common/modules/rg_psu.c
===================================================================
--- sonic-platform-modules-ruijie.orig/common/modules/rg_psu.c
+++ sonic-platform-modules-ruijie/common/modules/rg_psu.c
@@ -259,7 +259,7 @@ static int psu_detect(struct i2c_client
 	return 0;
 }
 
-static int psu_probe(struct i2c_client *client, const struct i2c_device_id *id)
+static int psu_probe(struct i2c_client *client)
 {
 	struct psu_data *data;
 	int status;
@@ -298,7 +298,7 @@ static int psu_probe(struct i2c_client *
 	return status;
 }
 
-static int psu_remove(struct i2c_client *client)
+static void psu_remove(struct i2c_client *client)
 {
 	switch (client->addr) {
 	case 0x50:
@@ -312,7 +312,6 @@ static int psu_remove(struct i2c_client
 	default:
 		break;
 	}
-	return 0;
 }
 
 static const struct i2c_device_id psu_id[] = {
